<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MsQuality: Calculation of QC metrics from mass spectrometry data • MsQuality</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MsQuality: Calculation of QC metrics from mass spectrometry data">
<meta property="og:description" content="MsQuality">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MsQuality</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MsQuality.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="MsQuality_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="MsQuality_files/pagedtable-1.1/js/pagedtable.js"></script><script src="MsQuality_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><script src="MsQuality_files/plotly-binding-4.10.2/plotly.js"></script><script src="MsQuality_files/typedarray-0.1/typedarray.min.js"></script><link href="MsQuality_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="MsQuality_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><link href="MsQuality_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="MsQuality_files/plotly-basic-2.11.1/plotly-basic-2.11.1.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MsQuality: Calculation of QC metrics from mass
spectrometry data</h1>
                        <h4 data-toc-skip class="author">Thomas Naake
and Johannes Rainer</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Meyerhofstrasse 1, 69117
Heidelberg/ EURAC Research, Viale Druso 1, 39100
Bolzano<br><div class="hidden name"><code>MsQuality.Rmd</code></div>

    </address>
</div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="sec-intro">Introduction<a class="anchor" aria-label="anchor" href="#sec-intro"></a>
</h2>
<p>Data quality assessment is an integral part of preparatory data
analysis to ensure sound biological information retrieval.</p>
<p>We present here the <code>MsQuality</code> package, which provides
functionality to calculate quality metrics for mass
spectrometry-derived, spectral data at the per-sample level.
<code>MsQuality</code> relies on the <a href="https://github.com/HUPO-PSI/mzQC" class="external-link"><code>mzQC</code></a> framework
of quality metrics defined by the Human Proteome Organization-Proteomics
Standards Intitiative (HUPO-PSI). These metrics quantify the quality of
spectral raw files using a controlled vocabulary. The package is
especially addressed towards users that acquire mass spectrometry data
on a large scale (e.g. data sets from clinical settings consisting of
several thousands of samples): while it is easier to control for
high-quality data acquisition in small-scale experiments, typically run
in one or few batches, clinical data sets are often acquired over longer
time frames and are prone to higher technical variation that is often
unnoticed. <code>MsQuality</code> tries to address this problem by
calculating metrics that can be stored along the spectral data sets (raw
files or feature-extracted data sets). <code>MsQuality</code>, thus,
facilitates the tracking of shifts in data quality and quantifies the
quality using multiple metrics. It should be thus easier to identify
samples that are of low quality (high-number of missing values,
termination of chromatographic runs, low instrument sensitivity,
etc.).</p>
<p>We would like to note here that these metrics only give an indication
of data quality, and, before removing indicated low-quality samples from
the analysis more advanced analytics, e.g. using the implemented
functionality and visualizations in the <code>MatrixQCvis</code>
package, should be scrutinized. Also, data quality should always be
regarded in the context of the sample type and experimental settings,
i.e. quality metrics should always be compared with regard to the sample
type, experimental setup, instrumentation, etc..</p>
<p>The <code>MsQuality</code> package allows to calculate low-level
quality metrics that require minimum information on mass spectrometry
data: retention time, m/z values, and associated intensities. The list
included in the <code>mzQC</code> framework is excessive, also including
metrics that rely on more high-level information, that might not be
readily accessible from .raw or .mzML files, e.g. pump pressure mean, or
rely on alignment results, e.g. retention time mean shift,
signal-to-noise ratio, precursor errors (ppm).</p>
<p>The <code>MsQuality</code> package is built upon the
<code>Spectra</code> and the <code>MsExperiment</code> package. Metrics
will be calculated based on the information stored in a
<code>Spectra</code> object and the respective <code>dataOrigin</code>
entries are used to distinguish between the mass spectral data of
multiple samples. The <code>MsExperiment</code> serves as a container to
store the mass spectral data of multiple samples. <code>MsQuality</code>
enables the user to calculate quality metrics both on
<code>Spectra</code> and <code>MsExperiment</code> objects.</p>
<p><code>MsQuality</code> can be used for any type of experiment that
can be represented as a <code>Spectra</code> or
<code>MsExperiment</code> object. This includes simple LC-MS data, DIA
or DDA-based data, ion mobility data or MS data in general. The tool can
thus be used for any type of targeted or untargeted metabolomics or
proteomics workflow. Also, we are not limited to data files in mzML
format, but, through <code>Spectra</code> and related
<code>MsBackend</code> packages, data can be imported from a large
variety of formats, including some raw vendor formats.</p>
<p>In this vignette, we will (i) create some exemplary
<code>Spectra</code> and <code>MsExperiment</code> objects, (ii)
calculate the quality metrics on these data sets, and (iii) visualize
some of the metrics.</p>
<div class="section level3">
<h3 id="alternative-software-for-data-quality-assessment">Alternative software for data quality assessment<a class="anchor" aria-label="anchor" href="#alternative-software-for-data-quality-assessment"></a>
</h3>
<p>Other <code>R</code> packages are available in Bioconductor that are
able to assess the quality of mass spectrometry data:</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/artMS.html" class="external-link"><code>artMS</code></a>
uses MaxQuant output and enables to calculate several QC metrics, e.g.
correlation matrix for technical replicates, calculation of total sum of
intensities in biological replicates, total peptide counts in biological
replicates, charge state distribution of PSMs identified in each
biological replicates, or MS1 scan counts in each biological
replicate.</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/MSstatsQC.html" class="external-link"><code>MSstatsQC</code></a>
and the visualization tool <a href="https://bioconductor.org/packages/release/bioc/html/MSstatsQCgui.html" class="external-link"><code>MSstatsQCgui</code></a>
require csv files in long format from spectral processing tools such as
Skyline and Panorama autoQC or <code>MSnbase</code> objects.
<code>MSstatsQC</code> enables to generate individual, moving range,
cumulative sum for mean, and/or cumulative sum for variability control
charts for each metric. Metrics can be any kind of user-defined metric
stored in the data columns for a given peptide, e.g. retention time and
peak area.</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/MQmetrics.html" class="external-link"><code>MQmetrics</code></a>
provides a pipeline to analyze the quality of proteomics data sets from
MaxQuant files and focuses on proteomics-/MaxQuant-specific metrics,
e.g. proteins identified, peptides identified, or proteins versus
peptide/protein ratio.</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/MatrixQCvis.html" class="external-link"><code>MatrixQCvis</code></a>
provides an interactive shiny-based interface to assess data quality at
various processing steps (normalization, transformation, batch
correction, and imputation) of rectangular matrices. The package
includes several diagnostic plots and metrics such as barplots of
intensity distributions, plots to visualize drifts, MA plots and
Hoeffding’s D value calculation, and dimension reduction plots and
provides specific tools to analyze data sets containing missing values
as commonly observed in mass spectrometry.</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/proBatch.html" class="external-link"><code>proBatch</code></a>
enables to assess batch effects in (prote)omics data sets and corrects
these batch effects in subsequent steps. Several tools to visualize data
quality are included in the <code>proBatch</code> packages, such as
barplots of intensity distributions, cluster and heatmap analysis tools,
and PCA dimension reduction plots. Additionally, <code>proBatch</code>
enables to assess diagnostics at the feature level, e.g. peptides or
spike-ins.</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install this package, start <code>R</code> and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## to install from Bioconductor</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MsQuality"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## to install from GitHub</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"tnaake/MsQuality"</span><span class="op">)</span></span></code></pre></div>
<p>This will install this package and all eventually missing
dependencies.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="questions-and-bugs">Questions and bugs<a class="anchor" aria-label="anchor" href="#questions-and-bugs"></a>
</h2>
<p><code>MsQuality</code> is currently under active development. If you
discover any bugs, typos or develop ideas of improving
<code>MsQuality</code> feel free to raise an issue via <a href="https://github.com/tnaake/MsQuality" class="external-link">GitHub</a> or send a mail to
the developer.</p>
</div>
<div class="section level2">
<h2 id="create-spectra-and-msexperiment-objects">Create <code>Spectra</code> and <code>MsExperiment</code>
objects<a class="anchor" aria-label="anchor" href="#create-spectra-and-msexperiment-objects"></a>
</h2>
<p>Load the <code>Spectra</code> package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/RforMassSpectrometry/Spectra" class="external-link">"Spectra"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/RforMassSpectrometry/MsExperiment" class="external-link">"MsExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://www.github.com/tnaake/MsQuality/" class="external-link">"MsQuality"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="create-spectra-and-msexperiment-objects-from-mzml-files">Create <code>Spectra</code> and <code>MsExperiment</code> objects
from mzML files<a class="anchor" aria-label="anchor" href="#create-spectra-and-msexperiment-objects-from-mzml-files"></a>
</h3>
<p>There are several options available to create a <code>Spectra</code>
object. One way, as outlined in the vignette of the <em><a href="https://bioconductor.org/packages/3.17/Spectra" class="external-link">Spectra</a></em>
package is by specifying the location of mass spectrometry raw files in
<code>mzML</code>, <code>mzXML</code> or <code>CDF</code> format and
using the <code>MsBackendMzR</code> backend. Here we load the example
files from the <code>sciex</code> data set of the <code>msdata</code>
package and create a <code>Spectra</code> object from the two provided
<code>mzML</code> files. The example is taken from the
<code>Spectra</code> vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## this example is taken from the Spectra vignette</span></span>
<span><span class="va">fls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"sciex"</span>, package <span class="op">=</span> <span class="st">"msdata"</span><span class="op">)</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">sps_sciex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/Spectra.html" class="external-link">Spectra</a></span><span class="op">(</span><span class="va">fls</span>, backend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Spectra/man/MsBackend.html" class="external-link">MsBackendMzR</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The data set consists of a single sample measured in two different
injections to the same LC-MS setup. An empty instance of an
<code>MsExperiment</code> object is created and populated with
information on the samples by assigning data on the samples
(<code>sampleData</code>), information on the <code>mzML</code> files
(<code>MsExperimentFiles</code>) and spectral information
(<code>spectra</code>). In a last step, using
<code>linkSampleData</code>, the relationships between the samples and
the spectral information are defined.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## this example is taken from the Spectra vignette</span></span>
<span><span class="va">lmse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MsExperiment/man/MsExperiment.html" class="external-link">MsExperiment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">sd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"QC1"</span>, <span class="st">"QC2"</span><span class="op">)</span>,</span>
<span>    sample_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"QC Pool"</span>, <span class="st">"QC Pool"</span><span class="op">)</span>,</span>
<span>    injection_idx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MsExperiment/man/MsExperiment.html" class="external-link">sampleData</a></span><span class="op">(</span><span class="va">lmse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sd</span></span>
<span><span class="co">## add mzML files to the experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MsExperiment/man/MsExperiment.html" class="external-link">experimentFiles</a></span><span class="op">(</span><span class="va">lmse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MsExperiment/man/MsExperimentFiles.html" class="external-link">MsExperimentFiles</a></span><span class="op">(</span>mzML_files <span class="op">=</span> <span class="va">fls</span><span class="op">)</span></span>
<span><span class="co">## add the Spectra object to the experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">spectra</a></span><span class="op">(</span><span class="va">lmse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sps_sciex</span></span>
<span><span class="co">## use linkSampleData to establish and define relationships between sample </span></span>
<span><span class="co">## annotations and MS data</span></span>
<span><span class="va">lmse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MsExperiment/man/MsExperiment.html" class="external-link">linkSampleData</a></span><span class="op">(</span><span class="va">lmse</span>, with <span class="op">=</span> <span class="st">"experimentFiles.mzML_file"</span>,</span>
<span>    sampleIndex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, withIndex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-spectra-and-msexperiment-objects-from-feature-extracted-intensity-tables">Create <code>Spectra</code> and <code>MsExperiment</code> objects
from (feature-extracted) intensity tables<a class="anchor" aria-label="anchor" href="#create-spectra-and-msexperiment-objects-from-feature-extracted-intensity-tables"></a>
</h3>
<p>Another common approach is the creation of <code>Spectra</code>
objects from a <code>DataFrame</code>s using the
<code>MsBackendDataFrame</code> backend.</p>
<p>We will use here the data set of <span class="citation">Lee et al.
(2019)</span>, that contains metabolite level information measured by
reverse phase liquid chromatography (RPLC) coupled to mass spectrometry
and hydrophilic interaction liquid chromatography (HILIC) coupled to
mass spectrometry (derived from the file
<code>STables - rev1.xlsx</code> in the Supplementary Information).</p>
<p>In a separate step (see documentation for
<code>Lee2019_meta_vals</code> and <code>Lee2019</code>), we have
created a list containing <code>Spectra</code> objects for each samples
(objects <code>sps_l_rplc</code> and <code>sps_l_hilic</code>) and
<code>MsExperiment</code> objects containing the data of all samples
(objects <code>msexp_rplc</code> and <code>msexp_hilic</code>). We will
load here these objects:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Lee2019"</span>, package <span class="op">=</span> <span class="st">"MsQuality"</span><span class="op">)</span></span></code></pre></div>
<p>The final data set contains 541 paired samples (i.e. 541 samples
derived from RPLC and 541 samples derived from HILIC).</p>
<p>We will combine the <code>sps_rplc</code> and <code>sps_hilic</code>
objects in the following and calculate on this combined document the
metrics.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sps_comb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sps_rplc</span>, <span class="va">sps_hilic</span><span class="op">)</span></span></code></pre></div>
<p>The most important function to assess the data quality and to
calculate the metrics is the <code>calculateMetrics</code> function. The
function takes a <code>Spectra</code> or <code>MsExperiment</code>
object as input, a character vector of metrics to be calculated, and,
optionally a list of parameters passed to the quality metrics
functions.</p>
</div>
</div>
<div class="section level2">
<h2 id="calculating-the-quality-metrics-on-spectra-and-msexperiment-objects">Calculating the quality metrics on <code>Spectra</code> and
<code>MsExperiment</code> objects<a class="anchor" aria-label="anchor" href="#calculating-the-quality-metrics-on-spectra-and-msexperiment-objects"></a>
</h2>
<p>Currently, the following metrics are included:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/qualityMetrics.html">qualityMetrics</a></span><span class="op">(</span><span class="va">sps_comb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "chromatographyDuration"             "ticQuartersRtFraction"             </span></span>
<span><span class="co">##  [3] "rtOverMsQuarters"                   "ticQuartileToQuartileLogRatio"     </span></span>
<span><span class="co">##  [5] "numberSpectra"                      "numberEmptyScans"                  </span></span>
<span><span class="co">##  [7] "medianPrecursorMz"                  "rtIqr"                             </span></span>
<span><span class="co">##  [9] "rtIqrRate"                          "areaUnderTic"                      </span></span>
<span><span class="co">## [11] "areaUnderTicRtQuantiles"            "extentIdentifiedPrecursorIntensity"</span></span>
<span><span class="co">## [13] "medianTicRtIqr"                     "medianTicOfRtRange"                </span></span>
<span><span class="co">## [15] "mzAcquisitionRange"                 "rtAcquisitionRange"                </span></span>
<span><span class="co">## [17] "precursorIntensityRange"            "precursorIntensityQuartiles"       </span></span>
<span><span class="co">## [19] "precursorIntensityMean"             "precursorIntensitySd"              </span></span>
<span><span class="co">## [21] "msSignal10xChange"                  "ratioCharge1over2"                 </span></span>
<span><span class="co">## [23] "ratioCharge3over2"                  "ratioCharge4over2"                 </span></span>
<span><span class="co">## [25] "meanCharge"                         "medianCharge"</span></span></code></pre>
<div class="section level3">
<h3 id="list-of-included-metrics">List of included metrics<a class="anchor" aria-label="anchor" href="#list-of-included-metrics"></a>
</h3>
<p>The following list gives a brief explanation for these metrics.
Further information may be found at the <a href="https://github.com/HUPO-PSI/mzQC" class="external-link">HUPO-PSI mzQC project page</a>
or in the respective help file for the quality metric (accessible by
e.g. entering <code><a href="../reference/chromatographyDuration.html">?chromatographyDuration</a></code> to the R console). We
also give explanation on how the metric is calculated in
<code>MsQuality</code>. Currently, all quality metrics can be calculated
for both <code>Spectra</code> and <code>MsExperiment</code> objects.</p>
<ul>
<li>
<em>chromatographyDuration</em>, <strong>chromatography
duration</strong> (MS:4000053), “The retention time duration of the
chromatography in seconds.” [PSI:MS]; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the retention time associated to the <code>Spectra</code> object is
obtained,</li>
<li>the maximum and the minimum of the retention time is obtained,</li>
<li>the difference between the maximum and the minimum is calculated and
returned.</li>
</ol>
</li>
<li>
<em>ticQuartersRtFraction</em>, <strong>TIC quarters RT
fraction</strong> (MS:4000054), “The interval when the respective
quarter of the TIC accumulates divided by retention time duration.”
[PSI:MS]; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is ordered according to the
retention time,<br>
</li>
<li>the cumulative sum of the ion count is calculated (TIC),</li>
<li>the quantiles are calculated according to the <code>probs</code>
argument, e.g. when <code>probs</code> is set to
<code>c(0, 0.25, 0.5, 0.75, 1)</code> the 0%, 25%, 50%, 75%, and 100%
quantile is calculated,</li>
<li>the retention time/relative retention time (retention time divided
by the total run time taking into account the minimum retention time) is
calculated,</li>
<li>the (relative) duration of the LC run after which the cumulative TIC
exceeds (for the first time) the respective quantile of the cumulative
TIC is calculated and returned.</li>
</ol>
</li>
<li>
<em>rtOverMsQuarters</em>, <strong>MS1 quarter RT fraction</strong>
(MS:4000055), “The interval used for acquisition of the first, second,
third, and fourth quarter of all MS1 events divided by retention time
duration.” [PSI:MS], <code>msLevel = 1L</code>; The metric is calculated
as follows:
<ol style="list-style-type: decimal">
<li>the retention time duration of the whole <code>Spectra</code> object
is determined (taking into account all the MS levels),</li>
<li>the <code>Spectra</code> object is filtered according to the MS
level and subsequently ordered according to the retention time<br>
</li>
<li>the MS events are split into four (approximately) equal parts,</li>
<li>the relative retention time is calculated (using the retention time
duration from (1) and taking into account the minimum retention
time),</li>
<li>the relative retention time values associated to the MS event parts
are returned.</li>
</ol>
</li>
<li>
<em>rtOverMsQuarters</em>, <strong>MS2 quarter RT fraction</strong>
(MS:4000056), “The interval used for acquisition of the first, second,
third, and fourth quarter of all MS2 events divided by retention time
duration.” [PSI:MS], <code>msLevel = 2L</code>; The metric is calculated
as follows:
<ol style="list-style-type: decimal">
<li>the retention time duration of the whole <code>Spectra</code> object
is determined (taking into account all the MS levels),</li>
<li>the <code>Spectra</code> object is filtered according to the MS
level and subsequently ordered according to the retention time<br>
</li>
<li>the MS events are split into four (approximately) equal parts,</li>
<li>the relative retention time is calculated (using the retention time
duration from (1) and taking into account the minimum retention
time),</li>
<li>the relative retention time values associated to the MS event parts
are returned.</li>
</ol>
</li>
<li>
<em>ticQuartileToQuartileLogRatio</em>, <strong>MS1 TIC-change
quartile ratios</strong> (MS:4000057), ““The log ratios of successive
TIC-change quartiles. The TIC changes are the list of MS1 total ion
current (TIC) value changes from one to the next scan, produced when
each MS1 TIC is subtracted from the preceding MS1 TIC. The metric’s
value triplet represents the log ratio of the TIC-change Q2 to Q1, Q3 to
Q2, TIC-change-max to Q3” [PSI:MS], <code>mode = "TIC_change"</code>,
<code>relativeTo = "previous"</code>, <code>msLevel = 1L</code>; The
metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the TIC (<code>ionCount</code>) of the <code>Spectra</code> object
is calculated per scan event (with spectra ordered by retention
time),</li>
<li>the differences between TIC values are calculated between subsequent
scan events,</li>
<li>the ratios between the 25%, 50%, 75%, and 100% quantile to the 25%
quantile of the values of (2) are calculated,</li>
<li>the <code>log</code> values of the ratios are returned.</li>
</ol>
</li>
<li>
<em>ticQuartileToQuartileLogRatio</em>, <strong>MS1 TIC quartile
ratios</strong> (MS:4000058), “The log ratios of successive TIC
quartiles. The metric’s value triplet represents the log ratios of
TIC-Q2 to TIC-Q1, TIC-Q3 to TIC-Q2, TIC-max to TIC-Q3.” [PSI:MS],
<code>mode = "TIC"</code>, <code>relativeTo = "previous"</code>,
<code>msLevel = 1L</code>; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the TIC (<code>ionCount</code>) of the <code>Spectra</code> object
is calculated per scan event (with spectra ordered by retention
time),</li>
<li>the TIC values between subsequent scan events are taken as they
are,</li>
<li>the ratios between the 25%, 50%, 75%, and 100% quantile to the 25%
quantile of the values of (2) are calculated.</li>
<li>The <code>log</code> values of the ratios are returned.</li>
</ol>
</li>
<li>
<em>numberSpectra</em>, <strong>number of MS1 spectra</strong>
MS:4000059), “The number of MS1 events in the run.” [PSI:MS],
<code>msLevel = 1L</code>; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the number of the spectra are obtained (<code>length</code> of
<code>Spectra</code>) and returned.</li>
</ol>
</li>
<li>
<em>numberSpectra</em>, <strong>number of MS2 spectra</strong>
(MS:4000060), “The number of MS2 events in the run.” [PSI:MS],
<code>msLevel = 2L</code>; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the number of the spectra are obtained (<code>length</code> of
<code>Spectra</code>) and returned.</li>
</ol>
</li>
<li>
<em>mzAcquisitionRange</em>, <strong>m/z acquisition range</strong>
(MS:4000069), “Upper and lower limit of m/z precursor values at which
MSn spectra are recorded.” [PSI:MS]; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the m/z values of the peaks within the <code>Spectra</code> object
are obtained,</li>
<li>the minimum and maximum m/z values are obtained and returned.</li>
</ol>
</li>
<li>
<em>rtAcquisitionRange</em>, <strong>retention time acquisition
range</strong> (MS:4000070), “Upper and lower limit of retention time at
which spectra are recorded.” [PSI:MS]; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the retention time values of the features within the
<code>Spectra</code> object are obtained,</li>
<li>the minimum and maximum retention time values are obtained and
returned.</li>
</ol>
</li>
<li>
<em>msSignal10xChange</em>, <strong>MS1 signal jump (10x)
count</strong> (MS:4000097), “The number of times where MS1 TIC
increased more than 10-fold between adjacent MS1 scans. An unusual high
count of signal jumps or falls can indicate ESI stability issues.”
[PSI:MS], <code>change = "jump"</code>, <code>msLevel = 1L</code>; The
metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the intensity values of the features are obtained via the ion
count,<br>
</li>
<li>the signal jumps/declines of the intensity values with the two
subsequent intensity values is calculated,</li>
<li>the signal jumps by a factor of ten or more are counted and
returned.</li>
</ol>
</li>
<li>
<em>msSignal10xChange</em>, <strong>MS1 signal fall (10x)
count</strong> (MS:4000098), “The number of times where MS1 TIC
decreased more than 10-fold between adjacent MS1 scans. An unusual high
count of signal jumps or falls can indicate ESI stability issues.”
[PSI:MS], <code>change = "fall"</code>, <code>msLevel = 1L</code>; The
metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the intensity values of the features are obtained via the ion
count,</li>
<li>the signal jumps/declines of the intensity values with the two
subsequent intensity values is calculated,</li>
<li>the signal declines by a factor of ten or more are counted and
returned.</li>
</ol>
</li>
<li>
<em>numberEmptyScans</em>, <strong>number of empty MS1
scans</strong> (MS:4000099), “Number of MS1 scans where the scans’ peaks
intensity sums to 0 (i.e. no peaks or only 0-intensity peaks).”
[PSI:MS], <code>msLevel = 1L</code>; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensities per entry are obtained,</li>
<li>the number of intensity entries that are <code>NULL</code>,
<code>NA</code>, or that have a sum of <code>0</code> are obtained and
returned.</li>
</ol>
</li>
<li>
<em>numberEmptyScans</em>, <strong>number of empty MS2
scans</strong> (MS:4000100), “Number of MS2 scans where the scans’ peaks
intensity sums to 0 (i.e. no peaks or only 0-intensity peaks).”
[PSI:MS], <code>msLevel = 2L</code>; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensities per entry are obtained,</li>
<li>the number of intensity entries that are <code>NULL</code>,
<code>NA</code>, or that have a sum of <code>0</code> are obtained and
returned.</li>
</ol>
</li>
<li>
<em>numberEmptyScans</em>, <strong>number of empty MS3
scans</strong> (MS:4000101), “Number of MS3 scans where the scans’ peaks
intensity sums to 0 (i.e. no peaks or only 0-intensity peaks).”
[PSI:MS], <code>msLevel = 3L</code>; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensities per entry are obtained,</li>
<li>the number of intensity entries that are <code>NULL</code>,
<code>NA</code>, or that have a sum of <code>0</code> are obtained and
returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityQuartiles</em>, <strong>MS2 precursor
intensity distribution Q1, Q2, Q3</strong> (MS:4000116), “From the
distribution of MS2 precursor intensities, the quartiles Q1, Q2, Q3. The
intensity distribution of the precursors informs about the dynamic range
of the acquisition.” [PSI:MS], <code>identificationLevel = "all"</code>;
The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the 25%, 50%, and 75% quantile of the precursor intensity values are
obtained (<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityMean</em>, <strong>MS2 precursor intensity
distribution mean</strong> (MS:4000117), “From the distribution of MS2
precursor intensities, the mean. The intensity distribution of the
precursors informs about the dynamic range of the acquisition.”
[PSI:MS], <code>identificationLevel = "all"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the mean of the precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensitySd</em>, <strong>MS2 precursor intensity
distribution sigma</strong> (MS:4000118), “From the distribution of MS2
precursor intensities, the sigma value. The intensity distribution of
the precursors informs about the dynamic range of the acquisition.”
[PSI:MS], <code>identificationLevel = "all"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the standard deviation of precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>medianPrecursorMz</em>, <strong>MS2 precursor median m/z of
identified quantification data points</strong> (MS:4000152), “Median m/z
value for MS2 precursors of all quantification data points after
user-defined acceptance criteria are applied. These data points may be
for example XIC profiles, isotopic pattern areas, or reporter ions (see
MS:1001805). The used type should be noted in the metadata or analysis
methods section of the recording file for the respective run. In case of
multiple acceptance criteria (FDR) available in proteomics, PSM-level
FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>,
<code>msLevel = 1L</code>; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor m/z values are obtained,</li>
<li>the median value is returned (<code>NA</code>s are removed).</li>
</ol>
</li>
<li>
<em>rtIqr</em>, <strong>interquartile RT period for identified
quantification data points</strong> (MS:4000153), “The interquartile
retention time period, in seconds, for all quantification data points
after user-defined acceptance criteria are applied over the complete
run. Longer times indicate better chromatographic separation. These data
points may be for example XIC profiles, isotopic pattern areas, or
reporter ions (see MS:1001805). The used type should be noted in the
metadata or analysis methods section of the recording file for the
respective run. In case of multiple acceptance criteria (FDR) available
in proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the retention time values are obtained,</li>
<li>the interquartile range is obtained from the values and returned
(<code>NA</code> values are removed).</li>
</ol>
</li>
<li>
<em>rtIqrRate</em>, <strong>rate of the interquartile RT period for
identified quantification data points</strong> (MS:4000154), “The rate
of identified quantification data points for the interquartile retention
time period, in identified quantification data points per second. Higher
rates indicate efficient sampling and identification. These data points
may be for example XIC profiles, isotopic pattern areas, or reporter
ions (see MS:1001805). The used type should be noted in the metadata or
analysis methods section of the recording file for the respective run.
In case of multiple acceptance criteria (FDR) available in proteomics,
PSM-level FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the retention time values are obtained,</li>
<li>the 25% and 75% quantiles are obtained from the retention time
values (<code>NA</code> values are removed),</li>
<li>the number of eluted features between this 25% and 75% quantile is
calculated,</li>
<li>the number of features is divided by the interquartile range of the
retention time and returned.</li>
</ol>
</li>
<li>
<em>areaUnderTic</em>, <strong>area under TIC</strong> (MS:4000155),
“The area under the total ion chromatogram.” [PSI:MS]; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the sum of the ion counts are obtained and returned.</li>
</ol>
</li>
<li>
<em>areaUnderTicRtQuantiles</em>, <strong>area under TIC RT
quantiles</strong> (MS:4000156), “The area under the total ion
chromatogram of the retention time quantiles. Number of quantiles are
given by the n-tuple.” [PSI:MS]; The metric is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the <code>Spectra</code> object is ordered according to the
retention time,</li>
<li>the 0%, 25%, 50%, 75%, and 100% quantiles of the retention time
values are obtained,</li>
<li>the ion count of the intervals between the 0%/25%, 25%/50%, 50%/75%,
and 75%/100% are obtained,<br>
</li>
<li>the ion counts of the intervals are summed (TIC) and the values
returned.</li>
</ol>
</li>
<li>
<em>extentIdentifiedPrecursorIntensity</em>, <strong>extent of
identified MS2 precursor intensity</strong> (MS:4000157), “Ratio of 95th
over 5th percentile of MS2 precursor intensity for all quantification
data points after user-defined acceptance criteria are applied. Can be
used to approximate the dynamic range of signal. The used type of
identification should be noted in the metadata or analysis methods
section of the recording file for the respective run. In case of
multiple acceptance criteria (FDR) available in proteomics, PSM-level
FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensities of the precursor ions are obtained,</li>
<li>the 5% and 95% quantile of these intensities are obtained (1NA1
values are removed),</li>
<li>the ratio between the 95% and the 5% intensity quantile is
calculated and returned.</li>
</ol>
</li>
<li>
<em>medianTicRtIqr</em>, <strong>median of TIC values in the RT
range in which the middle half of quantification data points are
identified</strong> (MS:4000158), “Median of TIC values in the RT range
in which half of quantification data points are identified (RT values of
Q1 to Q3 of identifications). These data points may be for example XIC
profiles, isotopic pattern areas, or reporter ions (see MS:1001805). The
used type should be noted in the metadata or analysis methods section of
the recording file for the respective run. In case of multiple
acceptance criteria (FDR) available in proteomics, PSM-level FDR should
be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the <code>Spectra</code> object is ordered according to the
retention time,</li>
<li>the features between the 1st and 3rd quartile are obtained (half of
the features that are present in the <code>Spectra</code> object),</li>
<li>the ion count of the features within the 1st and 3rd quartile is
obtained,</li>
<li>the median value of the ion count is calculated (<code>NA</code>
values are removed) and the median value is returned.</li>
</ol>
</li>
<li>
<em>medianTicOfRtRange</em>, <strong>median of TIC values in the
shortest RT range in which half of the quantification data points are
identified</strong> (MS:4000159), “Median of TIC values in the shortest
RT range in which half of the quantification data points are identified.
These data points may be for example XIC profiles, isotopic pattern
areas, or reporter ions (see MS:1001805). The used type should be noted
in the metadata or analysis methods section of the recording file for
the respective run. In case of multiple acceptance criteria (FDR)
available in proteomics, PSM-level FDR should be used for better
comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the <code>Spectra</code> object is ordered according to the
retention time,<br>
</li>
<li>the number of features in the <code>Spectra</code> object is
obtained and the number for half of the features is calculated,</li>
<li>iterate through the features (always by taking the neighbouring half
of features) and calculate the retention time range of the set of
features,</li>
<li>retrieve the set of features with the minimum retention time
range,</li>
<li>calculate from the set of (5) the median TIC (<code>NA</code> values
are removed) and return it.</li>
</ol>
</li>
<li>
<em>precursorIntensityRange</em>, <strong>MS2 precursor intensity
range</strong> (MS:4000160), “Minimum and maximum MS2 precursor
intensity recorded. The intensity range of the precursors informs about
the dynamic range of the acquisition.” [PSI:MS]; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the minimum and maximum precursor intensity values are obtained and
returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityQuartiles</em>, <strong>identified MS2
precursor intensity distribution Q1, Q2, Q3</strong> (MS:4000161), “From
the distribution of identified MS2 precursor intensities, the quartiles
Q1, Q2, Q3. The intensity distribution of the precursors informs about
the dynamic range of the acquisition in relation to identifiability. The
used type of identification should be noted in the metadata or analysis
methods section of the recording file for the respective run. In case of
multiple acceptance criteria (FDR) available in proteomics, PSM-level
FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the 25%, 50%, and 75% quantile of the precursor intensity values are
obtained (<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityQuartiles</em>, <strong>unidentified MS2
precursor intensity distribution Q1, Q2, Q3</strong> (MS:4000162), “From
the distribution of unidentified MS2 precursor intensities, the
quartiles Q1, Q2, Q3. The intensity distribution of the precursors
informs about the dynamic range of the acquisition in relation to
identifiability. The used type of identification should be noted in the
metadata or analysis methods section of the recording file for the
respective run. In case of multiple acceptance criteria (FDR) available
in proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "unidentified"</code>; The metric
is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the 25%, 50%, and 75% quantile of the precursor intensity values are
obtained (<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityMean</em>, <strong>identified MS2 precursor
intensity distribution mean</strong> (MS:4000163), “From the
distribution of identified MS2 precursor intensities, the mean. The
intensity distribution of the identified precursors informs about the
dynamic range of the acquisition in relation to identifiability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the mean of the precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensityMean</em>, <strong>unidentified MS2 precursor
intensity distribution mean</strong> (MS:4000164), “From the
distribution of unidentified MS2 precursor intensities, the mean. The
intensity distribution of the unidentified precursors informs about the
dynamic range of the acquisition in relation to identifiability.”
[PSI:MS], <code>identificationLevel = "unidentified"</code>; The metric
is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the mean of the precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensitySd</em>, <strong>identified MS2 precursor
intensity distribution sigma</strong> (MS:4000165), “From the
distribution of identified MS2 precursor intensities, the sigma value.
The intensity distribution of the precursors informs about the dynamic
range of the acquisition in relation to identifiability. The used type
of identification should be noted in the metadata or analysis methods
section of the recording file for the respective run. In case of
multiple acceptance criteria (FDR) available in proteomics, PSM-level
FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the standard deviation of precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>precursorIntensitySD</em>, <strong>unidentified MS2 precursor
intensity distribution sigma</strong> (MS:4000166), “From the
distribution of unidentified MS2 precursor intensities, the sigma value.
The intensity distribution of the precursors informs about the dynamic
range of the acquisition in relation to identifiability. The used type
of identification should be noted in the metadata or analysis methods
section of the recording file for the respective run. In case of
multiple acceptance criteria (FDR) available in proteomics, PSM-level
FDR should be used for better comparability.” [PSI:MS],
<code>identificationLevel = "unidentified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the intensity of the precursor ions within the <code>Spectra</code>
object are obtained,</li>
<li>the standard deviation of precursor intensity values is obtained
(<code>NA</code> values are removed) and returned.</li>
</ol>
</li>
<li>
<em>ratioCharge1over2</em>, <strong>ratio of 1+ over 2+ of all MS2
known precursor charges</strong> (MS:4000167), “The ratio of 1+ over 2+
MS2 precursor charge count of all spectra. High ratios of 1+/2+ MS2
precursor charge count may indicate inefficient ionization.” [PSI:MS],
<code>identificationLevel = "all"</code>; The metric is calculated as
follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the number of precursors with charge 1+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>ratioCharge1over2</em>, <strong>ratio of 1+ over 2+ of
identified MS2 known precursor charges</strong> (MS:4000168), “The ratio
of 1+ over 2+ MS2 precursor charge count of identified spectra. High
ratios of 1+/2+ MS2 precursor charge count may indicate inefficient
ionization. The used type of identification should be noted in the
metadata or analysis methods section of the recording file for the
respective run. In case of multiple acceptance criteria (FDR) available
in proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the number of precursors with charge 1+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>ratioCharge3over2</em>, <strong>ratio of 3+ over 2+ of all MS2
known precursor charges</strong> (MS:4000169), “The ratio of 3+ over 2+
MS2 precursor charge count of all spectra. Higher ratios of 3+/2+ MS2
precursor charge count may preferentially favor longer e.g. peptides.”
[PSI:MS], <code>identificationLevel = "all"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the precursor charge is obtained,<br>
</li>
<li>the number of precursors with charge 3+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>ratioCharge3over2</em>, <strong>ratio of 3+ over 2+ of
identified MS2 known precursor charges</strong> (MS:4000170), “The ratio
of 3+ over 2+ MS2 precursor charge count of identified spectra. Higher
ratios of 3+/2+ MS2 precursor charge count may preferentially favor
longer e.g. peptides. The used type of identification should be noted in
the metadata or analysis methods section of the recording file for the
respective run. In case of multiple acceptance criteria (FDR) available
in proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,<br>
</li>
<li>the precursor charge is obtained,<br>
</li>
<li>the number of precursors with charge 3+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>ratioCharge4over2</em>, <strong>ratio of 4+ over 2+ of all MS2
known precursor charges</strong> (MS:4000171), “The ratio of 4+ over 2+
MS2 precursor charge count of all spectra. Higher ratios of 4+/2+ MS2
precursor charge count may preferentially favor longer e.g. peptides.”
[PSI:MS], <code>identificationLevel = "all"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the number of precursors with charge 4+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>ratioCharge4over2</em>, <strong>ratio of 4+ over 2+ of
identified MS2 known precursor charges</strong> (MS:4000172), “The ratio
of 4+ over 2+ MS2 precursor charge count of identified spectra. Higher
ratios of 4+/2+ MS2 precursor charge count may preferentially favor
longer e.g. peptides. The used type of identification should be noted in
the metadata or analysis methods section of the recording file for the
respective run. In case of multiple acceptance criteria (FDR) available
in proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the number of precursors with charge 4+ is divided by the number of
precursors with charge 2+ and the ratio is returned.</li>
</ol>
</li>
<li>
<em>meanCharge</em>, <strong>mean MS2 precursor charge in all
spectra</strong> (MS:4000173), “Mean MS2 precursor charge in all
spectra” [PSI:MS], <code>identificationLevel = "all"</code>; The metric
is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the mean of the precursor charge values is calculated and
returned.</li>
</ol>
</li>
<li>
<em>meanCharge</em>, <strong>mean MS2 precursor charge in identified
spectra</strong> (MS:4000174), “Mean MS2 precursor charge in identified
spectra. The used type of identification should be noted in the metadata
or analysis methods section of the recording file for the respective
run. In case of multiple acceptance criteria (FDR) available in
proteomics, PSM-level FDR should be used for better comparability.”
[PSI:MS], <code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the mean of the precursor charge values is calculated and
returned.</li>
</ol>
</li>
<li>
<em>medianCharge</em>, <strong>median MS2 precursor charge in all
spectra</strong> (MS:4000175), “Median MS2 precursor charge in all
spectra” [PSI:MS], <code>identificationLevel = "all"</code>; The metric
is calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the median of the precursor charge values is calculated and
returned.</li>
</ol>
</li>
<li>
<em>medianCharge</em>, <strong>median MS2 precursor charge in
identified spectra</strong> (MS:4000176), “Median MS2 precursor charge
in identified spectra. The used type of identification should be noted
in the metadata or analysis methods section of the recording file for
the respective run. In case of multiple acceptance criteria (FDR)
available in proteomics, PSM-level FDR should be used for better
comparability.” [PSI:MS],
<code>identificationLevel = "identified"</code>; The metric is
calculated as follows:
<ol style="list-style-type: decimal">
<li>the <code>Spectra</code> object is filtered according to the MS
level,</li>
<li>the precursor charge is obtained,</li>
<li>the median of the precursor charge values is calculated and
returned.</li>
</ol>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="obtain-the-metrics-in-data-frame-format">Obtain the metrics in <code>data.frame</code>-format<a class="anchor" aria-label="anchor" href="#obtain-the-metrics-in-data-frame-format"></a>
</h3>
<p>The most important function to assess the data quality and to
calculate the metrics is the <code>calculateMetrics</code> function. The
function takes a <code>Spectra</code> or <code>MsExperiment</code>
object as input, a character vector of metrics to be calculated, a
Boolean value to the <code>filterEmptySpectra</code> argument, and,
optionally a list of parameters passed to the quality metrics
functions.</p>
<p>The <code>filterEmptySpectra</code> argument specifies if
zero-intensity,
<code>Inf</code>-intensity<code>or zero-length entries should be removed (</code>filterEmptySpectra
=
TRUE<code>).  By default, the entries are taken as they are (</code>filterEmptySpectra
=
FALSE<code>).  The argument can be set to</code>TRUE<code>to compute metrics that are close to the  implementation of the</code>QuaMeter<code>software. Prior to calculating the metrics,  the implementation of</code>QuaMeter<code>skips all spectra with</code>defaultArrayLength=0`
(in .mzML files) at any MS level.</p>
<p>When passing a <code>Spectra</code>/<code>MsExperiment</code> object
to the function, a <code>data.frame</code> returned by
<code>calculateMetrics</code> with the metrics specified by the argument
<code>metrics</code>. By default, <code>qualityMetrics(object)</code> is
taken to specify the calculation of quality metrics.
<code>calculateMetrics</code> also accepts a list of parameters passed
to the individual quality metrics functions. For each quality metrics
functions, the relevant parameters are selected based on the accepted
arguments.</p>
<p>Additional arguments can be given to the quality metrics functions.
For example, the function <code>ticQuartileToQuartileLogRatio</code>
function has the arguments <code>relativeTo</code>, <code>mode</code>,
and <code>msLevel</code>. <code>relativeTo</code> specifies to which
quantile the log TIC quantile is relatively related to (either to the
1st quantile or the respective previous one). <code>mode</code> (either
<code>"TIC_change"</code> or <code>"TIC"</code>) specifies if the
quantiles are taken from the changes between TICs of scan events or the
TICs directly. One <code>Spectra</code>/<code>MsExperiment</code> object
may also contain more than one <code>msLevel</code>, e.g. if it also
contains information on MS<span class="math inline">\(^2\)</span> or
MS<span class="math inline">\(^3\)</span> features. If the user adds the
arguments
<code>relativeTo = "Q1", mode = "TIC", msLevel = c(1L, 2L))</code>,
<code>ticQuartileToQuartileLogRatio</code> is run with the parameter
combinations
<code>relativeTo = "Q1", mode = "TIC", msLevel = c(1L, 2L)</code>.</p>
<p>The results based on these parameter combinations are returned and
the used parameters are returned as attributes to the returned
vector.</p>
<p>Here, we would like to calculate the metrics of <strong>all</strong>
included quality metrics functions (<code>qualityMetrics(object)</code>)
and additionally pass the parameter <code>relativeTo = "Q1"</code> and
<code>relativeTo = "previous"</code>. For computational reasons, we will
restrict the calculation of the metrics to the first<br>
sample and to RPLC samples.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## subset the Spectra objects</span></span>
<span><span class="va">sps_comb_subset</span> <span class="op">&lt;-</span> <span class="va">sps_comb</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Sample.1_"</span>, <span class="va">sps_comb</span><span class="op">$</span><span class="va">dataOrigin</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## for RPLC and HILIC</span></span>
<span><span class="va">metrics_sps_Q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateMetrics.html">calculateMetrics</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sps_comb_subset</span>,</span>
<span>    metrics <span class="op">=</span> <span class="fu"><a href="../reference/qualityMetrics.html">qualityMetrics</a></span><span class="op">(</span><span class="va">sps_comb_subset</span><span class="op">)</span>, filterEmptySpectra <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    relativeTo <span class="op">=</span> <span class="st">"Q1"</span>, msLevel <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span>
<span><span class="va">metrics_sps_Q1</span></span></code></pre></div>
<pre><code><span><span class="co">##                chromatographyDuration ticQuartersRtFraction.0%</span></span>
<span><span class="co">## Sample.1_RPLC                  18.214                        0</span></span>
<span><span class="co">## Sample.1_HILIC                 16.000                        0</span></span>
<span><span class="co">##                ticQuartersRtFraction.25% ticQuartersRtFraction.50%</span></span>
<span><span class="co">## Sample.1_RPLC                 0.08098166                0.08098166</span></span>
<span><span class="co">## Sample.1_HILIC                0.34375000                0.34375000</span></span>
<span><span class="co">##                ticQuartersRtFraction.75% ticQuartersRtFraction.100%</span></span>
<span><span class="co">## Sample.1_RPLC                  0.1495004                          1</span></span>
<span><span class="co">## Sample.1_HILIC                 0.5375000                          1</span></span>
<span><span class="co">##                rtOverMsQuarters.Quarter1 rtOverMsQuarters.Quarter2</span></span>
<span><span class="co">## Sample.1_RPLC                 0.02893379                0.08806413</span></span>
<span><span class="co">## Sample.1_HILIC                0.15625000                0.47500000</span></span>
<span><span class="co">##                rtOverMsQuarters.Quarter3 rtOverMsQuarters.Quarter4</span></span>
<span><span class="co">## Sample.1_RPLC                  0.3102009                         1</span></span>
<span><span class="co">## Sample.1_HILIC                 0.6216875                         1</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q2/Q1</span></span>
<span><span class="co">## Sample.1_RPLC                                  NaN</span></span>
<span><span class="co">## Sample.1_HILIC                                -Inf</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q3/Q1</span></span>
<span><span class="co">## Sample.1_RPLC                                  NaN</span></span>
<span><span class="co">## Sample.1_HILIC                                 NaN</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q4/Q1 numberSpectra</span></span>
<span><span class="co">## Sample.1_RPLC                                  NaN           190</span></span>
<span><span class="co">## Sample.1_HILIC                                 NaN           165</span></span>
<span><span class="co">##                numberEmptyScans medianPrecursorMz   rtIqr rtIqrRate</span></span>
<span><span class="co">## Sample.1_RPLC                 0            198.05 5.10125  18.42686</span></span>
<span><span class="co">## Sample.1_HILIC                0            179.10 7.44700  11.14543</span></span>
<span><span class="co">##                areaUnderTic areaUnderTicRtQuantiles.25%</span></span>
<span><span class="co">## Sample.1_RPLC     655624525                  25612593.2</span></span>
<span><span class="co">## Sample.1_HILIC    149945016                    927493.9</span></span>
<span><span class="co">##                areaUnderTicRtQuantiles.50% areaUnderTicRtQuantiles.75%</span></span>
<span><span class="co">## Sample.1_RPLC                    389734297                   233673968</span></span>
<span><span class="co">## Sample.1_HILIC                   100961764                    40996727</span></span>
<span><span class="co">##                areaUnderTicRtQuantiles.100% extentIdentifiedPrecursorIntensity</span></span>
<span><span class="co">## Sample.1_RPLC                       5692460                          36467.884</span></span>
<span><span class="co">## Sample.1_HILIC                      5460065                           8713.906</span></span>
<span><span class="co">##                medianTicRtIqr medianTicOfRtRange mzAcquisitionRange.min</span></span>
<span><span class="co">## Sample.1_RPLC        7496.006          13858.935                   60.1</span></span>
<span><span class="co">## Sample.1_HILIC       2195.400           2086.417                   73.0</span></span>
<span><span class="co">##                mzAcquisitionRange.max rtAcquisitionRange.min</span></span>
<span><span class="co">## Sample.1_RPLC                  1377.6                  0.986</span></span>
<span><span class="co">## Sample.1_HILIC                  784.1                  1.100</span></span>
<span><span class="co">##                rtAcquisitionRange.max precursorIntensityRange.min</span></span>
<span><span class="co">## Sample.1_RPLC                    19.2                    100.6492</span></span>
<span><span class="co">## Sample.1_HILIC                   17.1                    100.0895</span></span>
<span><span class="co">##                precursorIntensityRange.max precursorIntensityQuartiles.Q1</span></span>
<span><span class="co">## Sample.1_RPLC                    349282909                      1667.3911</span></span>
<span><span class="co">## Sample.1_HILIC                    92021751                       300.5038</span></span>
<span><span class="co">##                precursorIntensityQuartiles.Q2 precursorIntensityQuartiles.Q3</span></span>
<span><span class="co">## Sample.1_RPLC                        6746.195                       75003.90</span></span>
<span><span class="co">## Sample.1_HILIC                       2304.383                       33382.25</span></span>
<span><span class="co">##                precursorIntensityMean precursorIntensitySd msSignal10xChange</span></span>
<span><span class="co">## Sample.1_RPLC               3450655.4             26563228                56</span></span>
<span><span class="co">## Sample.1_HILIC               908757.7              7729451                47</span></span>
<span><span class="co">##                ratioCharge1over2 ratioCharge3over2 ratioCharge4over2 meanCharge</span></span>
<span><span class="co">## Sample.1_RPLC                NaN               NaN               NaN        NaN</span></span>
<span><span class="co">## Sample.1_HILIC               NaN               NaN               NaN        NaN</span></span>
<span><span class="co">##                medianCharge</span></span>
<span><span class="co">## Sample.1_RPLC            NA</span></span>
<span><span class="co">## Sample.1_HILIC           NA</span></span>
<span><span class="co">## attr(,"chromatographyDuration")</span></span>
<span><span class="co">## [1] "MS:4000053"</span></span>
<span><span class="co">## attr(,"names1")</span></span>
<span><span class="co">## [1] "Q1"</span></span>
<span><span class="co">## attr(,"names2")</span></span>
<span><span class="co">## [1] "Q2"</span></span>
<span><span class="co">## attr(,"names3")</span></span>
<span><span class="co">## [1] "Q3"</span></span>
<span><span class="co">## attr(,"names4")</span></span>
<span><span class="co">## [1] "100%"</span></span>
<span><span class="co">## attr(,"names5")</span></span>
<span><span class="co">## [1] "100%"</span></span>
<span><span class="co">## attr(,"ticQuartersRtFraction")</span></span>
<span><span class="co">## [1] "MS:4000054"</span></span>
<span><span class="co">## attr(,"rtOverMsQuarters")</span></span>
<span><span class="co">## [1] "MS:4000055"</span></span>
<span><span class="co">## attr(,"numberSpectra")</span></span>
<span><span class="co">## [1] "MS:4000059"</span></span>
<span><span class="co">## attr(,"numberEmptyScans")</span></span>
<span><span class="co">## [1] "MS:4000099"</span></span>
<span><span class="co">## attr(,"areaUnderTic")</span></span>
<span><span class="co">## [1] "MS:4000155"</span></span>
<span><span class="co">## attr(,"areaUnderTicRtQuantiles")</span></span>
<span><span class="co">## [1] "MS:4000156"</span></span>
<span><span class="co">## attr(,"mzAcquisitionRange")</span></span>
<span><span class="co">## [1] "MS:4000069"</span></span>
<span><span class="co">## attr(,"rtAcquisitionRange")</span></span>
<span><span class="co">## [1] "MS:4000070"</span></span>
<span><span class="co">## attr(,"precursorIntensityRange")</span></span>
<span><span class="co">## [1] "MS:4000160"</span></span>
<span><span class="co">## attr(,"precursorIntensityQuartiles")</span></span>
<span><span class="co">## [1] "MS:4000116"</span></span>
<span><span class="co">## attr(,"precursorIntensityMean")</span></span>
<span><span class="co">## [1] "MS:4000117"</span></span>
<span><span class="co">## attr(,"precursorIntensitySd")</span></span>
<span><span class="co">## [1] "MS:4000118"</span></span>
<span><span class="co">## attr(,"msSignal10xChange")</span></span>
<span><span class="co">## [1] "MS:4000097"</span></span>
<span><span class="co">## attr(,"ratioCharge1over2")</span></span>
<span><span class="co">## [1] "MS:4000167"</span></span>
<span><span class="co">## attr(,"ratioCharge3over2")</span></span>
<span><span class="co">## [1] "MS:4000169"</span></span>
<span><span class="co">## attr(,"ratioCharge4over2")</span></span>
<span><span class="co">## [1] "MS:4000171"</span></span>
<span><span class="co">## attr(,"meanCharge")</span></span>
<span><span class="co">## [1] "MS:4000173"</span></span>
<span><span class="co">## attr(,"medianCharge")</span></span>
<span><span class="co">## [1] "MS:4000175"</span></span>
<span><span class="co">## attr(,"relativeTo")</span></span>
<span><span class="co">## [1] "Q1"</span></span>
<span><span class="co">## attr(,"msLevel")</span></span>
<span><span class="co">## [1] 1</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metrics_sps_previous</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateMetrics.html">calculateMetrics</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sps_comb_subset</span>,</span>
<span>    metrics <span class="op">=</span> <span class="fu"><a href="../reference/qualityMetrics.html">qualityMetrics</a></span><span class="op">(</span><span class="va">sps_comb_subset</span><span class="op">)</span>, filterEmptySpectra <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    relativeTo <span class="op">=</span> <span class="st">"previous"</span>, msLevel <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span>
<span><span class="va">metrics_sps_previous</span></span></code></pre></div>
<pre><code><span><span class="co">##                chromatographyDuration ticQuartersRtFraction.0%</span></span>
<span><span class="co">## Sample.1_RPLC                  18.214                        0</span></span>
<span><span class="co">## Sample.1_HILIC                 16.000                        0</span></span>
<span><span class="co">##                ticQuartersRtFraction.25% ticQuartersRtFraction.50%</span></span>
<span><span class="co">## Sample.1_RPLC                 0.08098166                0.08098166</span></span>
<span><span class="co">## Sample.1_HILIC                0.34375000                0.34375000</span></span>
<span><span class="co">##                ticQuartersRtFraction.75% ticQuartersRtFraction.100%</span></span>
<span><span class="co">## Sample.1_RPLC                  0.1495004                          1</span></span>
<span><span class="co">## Sample.1_HILIC                 0.5375000                          1</span></span>
<span><span class="co">##                rtOverMsQuarters.Quarter1 rtOverMsQuarters.Quarter2</span></span>
<span><span class="co">## Sample.1_RPLC                 0.02893379                0.08806413</span></span>
<span><span class="co">## Sample.1_HILIC                0.15625000                0.47500000</span></span>
<span><span class="co">##                rtOverMsQuarters.Quarter3 rtOverMsQuarters.Quarter4</span></span>
<span><span class="co">## Sample.1_RPLC                  0.3102009                         1</span></span>
<span><span class="co">## Sample.1_HILIC                 0.6216875                         1</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q2/Q1</span></span>
<span><span class="co">## Sample.1_RPLC                                  NaN</span></span>
<span><span class="co">## Sample.1_HILIC                                -Inf</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q3/Q2</span></span>
<span><span class="co">## Sample.1_RPLC                             5.831233</span></span>
<span><span class="co">## Sample.1_HILIC                                 Inf</span></span>
<span><span class="co">##                ticQuartileToQuartileLogRatio.Q4/Q3 numberSpectra</span></span>
<span><span class="co">## Sample.1_RPLC                             8.915513           190</span></span>
<span><span class="co">## Sample.1_HILIC                            8.982638           165</span></span>
<span><span class="co">##                numberEmptyScans medianPrecursorMz   rtIqr rtIqrRate</span></span>
<span><span class="co">## Sample.1_RPLC                 0            198.05 5.10125  18.42686</span></span>
<span><span class="co">## Sample.1_HILIC                0            179.10 7.44700  11.14543</span></span>
<span><span class="co">##                areaUnderTic areaUnderTicRtQuantiles.25%</span></span>
<span><span class="co">## Sample.1_RPLC     655624525                  25612593.2</span></span>
<span><span class="co">## Sample.1_HILIC    149945016                    927493.9</span></span>
<span><span class="co">##                areaUnderTicRtQuantiles.50% areaUnderTicRtQuantiles.75%</span></span>
<span><span class="co">## Sample.1_RPLC                    389734297                   233673968</span></span>
<span><span class="co">## Sample.1_HILIC                   100961764                    40996727</span></span>
<span><span class="co">##                areaUnderTicRtQuantiles.100% extentIdentifiedPrecursorIntensity</span></span>
<span><span class="co">## Sample.1_RPLC                       5692460                          36467.884</span></span>
<span><span class="co">## Sample.1_HILIC                      5460065                           8713.906</span></span>
<span><span class="co">##                medianTicRtIqr medianTicOfRtRange mzAcquisitionRange.min</span></span>
<span><span class="co">## Sample.1_RPLC        7496.006          13858.935                   60.1</span></span>
<span><span class="co">## Sample.1_HILIC       2195.400           2086.417                   73.0</span></span>
<span><span class="co">##                mzAcquisitionRange.max rtAcquisitionRange.min</span></span>
<span><span class="co">## Sample.1_RPLC                  1377.6                  0.986</span></span>
<span><span class="co">## Sample.1_HILIC                  784.1                  1.100</span></span>
<span><span class="co">##                rtAcquisitionRange.max precursorIntensityRange.min</span></span>
<span><span class="co">## Sample.1_RPLC                    19.2                    100.6492</span></span>
<span><span class="co">## Sample.1_HILIC                   17.1                    100.0895</span></span>
<span><span class="co">##                precursorIntensityRange.max precursorIntensityQuartiles.Q1</span></span>
<span><span class="co">## Sample.1_RPLC                    349282909                      1667.3911</span></span>
<span><span class="co">## Sample.1_HILIC                    92021751                       300.5038</span></span>
<span><span class="co">##                precursorIntensityQuartiles.Q2 precursorIntensityQuartiles.Q3</span></span>
<span><span class="co">## Sample.1_RPLC                        6746.195                       75003.90</span></span>
<span><span class="co">## Sample.1_HILIC                       2304.383                       33382.25</span></span>
<span><span class="co">##                precursorIntensityMean precursorIntensitySd msSignal10xChange</span></span>
<span><span class="co">## Sample.1_RPLC               3450655.4             26563228                56</span></span>
<span><span class="co">## Sample.1_HILIC               908757.7              7729451                47</span></span>
<span><span class="co">##                ratioCharge1over2 ratioCharge3over2 ratioCharge4over2 meanCharge</span></span>
<span><span class="co">## Sample.1_RPLC                NaN               NaN               NaN        NaN</span></span>
<span><span class="co">## Sample.1_HILIC               NaN               NaN               NaN        NaN</span></span>
<span><span class="co">##                medianCharge</span></span>
<span><span class="co">## Sample.1_RPLC            NA</span></span>
<span><span class="co">## Sample.1_HILIC           NA</span></span>
<span><span class="co">## attr(,"chromatographyDuration")</span></span>
<span><span class="co">## [1] "MS:4000053"</span></span>
<span><span class="co">## attr(,"names1")</span></span>
<span><span class="co">## [1] "Q1"</span></span>
<span><span class="co">## attr(,"names2")</span></span>
<span><span class="co">## [1] "Q2"</span></span>
<span><span class="co">## attr(,"names3")</span></span>
<span><span class="co">## [1] "Q3"</span></span>
<span><span class="co">## attr(,"names4")</span></span>
<span><span class="co">## [1] "100%"</span></span>
<span><span class="co">## attr(,"names5")</span></span>
<span><span class="co">## [1] "100%"</span></span>
<span><span class="co">## attr(,"ticQuartersRtFraction")</span></span>
<span><span class="co">## [1] "MS:4000054"</span></span>
<span><span class="co">## attr(,"rtOverMsQuarters")</span></span>
<span><span class="co">## [1] "MS:4000055"</span></span>
<span><span class="co">## attr(,"ticQuartileToQuartileLogRatio")</span></span>
<span><span class="co">## [1] "MS:4000057"</span></span>
<span><span class="co">## attr(,"numberSpectra")</span></span>
<span><span class="co">## [1] "MS:4000059"</span></span>
<span><span class="co">## attr(,"numberEmptyScans")</span></span>
<span><span class="co">## [1] "MS:4000099"</span></span>
<span><span class="co">## attr(,"areaUnderTic")</span></span>
<span><span class="co">## [1] "MS:4000155"</span></span>
<span><span class="co">## attr(,"areaUnderTicRtQuantiles")</span></span>
<span><span class="co">## [1] "MS:4000156"</span></span>
<span><span class="co">## attr(,"mzAcquisitionRange")</span></span>
<span><span class="co">## [1] "MS:4000069"</span></span>
<span><span class="co">## attr(,"rtAcquisitionRange")</span></span>
<span><span class="co">## [1] "MS:4000070"</span></span>
<span><span class="co">## attr(,"precursorIntensityRange")</span></span>
<span><span class="co">## [1] "MS:4000160"</span></span>
<span><span class="co">## attr(,"precursorIntensityQuartiles")</span></span>
<span><span class="co">## [1] "MS:4000116"</span></span>
<span><span class="co">## attr(,"precursorIntensityMean")</span></span>
<span><span class="co">## [1] "MS:4000117"</span></span>
<span><span class="co">## attr(,"precursorIntensitySd")</span></span>
<span><span class="co">## [1] "MS:4000118"</span></span>
<span><span class="co">## attr(,"msSignal10xChange")</span></span>
<span><span class="co">## [1] "MS:4000097"</span></span>
<span><span class="co">## attr(,"ratioCharge1over2")</span></span>
<span><span class="co">## [1] "MS:4000167"</span></span>
<span><span class="co">## attr(,"ratioCharge3over2")</span></span>
<span><span class="co">## [1] "MS:4000169"</span></span>
<span><span class="co">## attr(,"ratioCharge4over2")</span></span>
<span><span class="co">## [1] "MS:4000171"</span></span>
<span><span class="co">## attr(,"meanCharge")</span></span>
<span><span class="co">## [1] "MS:4000173"</span></span>
<span><span class="co">## attr(,"medianCharge")</span></span>
<span><span class="co">## [1] "MS:4000175"</span></span>
<span><span class="co">## attr(,"relativeTo")</span></span>
<span><span class="co">## [1] "previous"</span></span>
<span><span class="co">## attr(,"msLevel")</span></span>
<span><span class="co">## [1] 1</span></span></code></pre>
<p>Alternatively, an <code>MsExperiment</code> object might be passed to
<code>calculateMetrics</code>. The function will iterate over the
samples (referring to rows in <code>sampleData(msexp))</code>) and
calculate the quality metrics on the corresponding
<code>Spectra</code>s.</p>
</div>
<div class="section level3">
<h3 id="obtain-the-metrics-in-mzqc-format">Obtain the metrics in <code>mzQC</code>-format<a class="anchor" aria-label="anchor" href="#obtain-the-metrics-in-mzqc-format"></a>
</h3>
<p>By default, a <code>data.frame</code> object containing the metric
values as entries are returned by the the function
<code>calculateMetrics</code>. Alternatively, the function also allows
the user to export the metrics in a format defined by the
<code>rmzqc</code> package by setting the argument <code>format</code>
to <code>"mzQC"</code> (default: <code>format = "data.frame"</code>). In
that case, only the metrics that comply to the <code>mzQC</code>
specification will be written to the returned object. The object can be
exported and validated using the functionality of the <code>rmzqc</code>
package (see the documentation of <code>rmzqc</code> for further
information).</p>
</div>
<div class="section level3">
<h3 id="remove-empty-spectra-prior-to-the-calculation">Remove empty spectra prior to the calculation<a class="anchor" aria-label="anchor" href="#remove-empty-spectra-prior-to-the-calculation"></a>
</h3>
<p>There are in total 541 samples respectively in the objects
<code>msexp_rplc</code> and <code>msexp_hilic</code>. To improve the
visualization and interpretability, we will only calculate the metrics
from the first 20 of these samples.</p>
<p>In this example here, we will remove zero-length and zero-intensity
entries prior to calculating the metrics. To do this, we set the
<code>filterEmptySpectra</code> argument to <code>TRUE</code> within the
<code>calculateMetrics</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## subset the MsExperiment objects</span></span>
<span><span class="va">msexp_rplc_subset</span> <span class="op">&lt;-</span> <span class="va">msexp_rplc</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span><span class="va">msexp_hilic_subset</span> <span class="op">&lt;-</span> <span class="va">msexp_hilic</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## define metrics</span></span>
<span><span class="va">metrics_sps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chromatographyDuration"</span>, <span class="st">"ticQuartersRtFraction"</span>, <span class="st">"rtOverMsQuarters"</span>,</span>
<span>    <span class="st">"ticQuartileToQuartileLogRatio"</span>, <span class="st">"numberSpectra"</span>, <span class="st">"medianPrecursorMz"</span>,</span>
<span>    <span class="st">"rtIqr"</span>, <span class="st">"rtIqrRate"</span>, <span class="st">"areaUnderTic"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## for RPLC-derived MsExperiment</span></span>
<span><span class="va">metrics_rplc_msexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateMetrics.html">calculateMetrics</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">msexp_rplc_subset</span>,</span>
<span>    metrics <span class="op">=</span> <span class="fu"><a href="../reference/qualityMetrics.html">qualityMetrics</a></span><span class="op">(</span><span class="va">msexp_rplc_subset</span><span class="op">)</span>, filterEmptySpectra <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    relativeTo <span class="op">=</span> <span class="st">"Q1"</span>, msLevel <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## for HILIC-derived MsExperiment</span></span>
<span><span class="va">metrics_hilic_msexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateMetrics.html">calculateMetrics</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">msexp_hilic_subset</span>,</span>
<span>    metrics <span class="op">=</span> <span class="fu"><a href="../reference/qualityMetrics.html">qualityMetrics</a></span><span class="op">(</span><span class="va">msexp_hilic_subset</span><span class="op">)</span>, filterEmptySpectra <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    relativeTo <span class="op">=</span> <span class="st">"Q1"</span>, msLevel <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span></code></pre></div>
<p>When passing an <code>MsExperiment</code> object to
<code>calculateMetrics</code> a <code>data.frame</code> object is
returned with the samples (derived from the rownames of
<code>sampleData(msexp)</code>) in the rows and the metrics in
columns.</p>
<p>We will show here the objects <code>metrics_rplc_msexp</code> and
<code>metrics_hilic_msexp</code></p>
<pre><code><span><span class="co">## [1] "metrics_rplc_msexp"</span></span></code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["chromatographyDuration"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.0%"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.25%"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.50%"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.75%"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.100%"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter1"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter2"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter3"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter4"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q2/Q1"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q3/Q1"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q4/Q1"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["numberSpectra"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["numberEmptyScans"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["medianPrecursorMz"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["rtIqr"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["rtIqrRate"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["areaUnderTic"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.25%"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.50%"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.75%"],"name":[22],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.100%"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["extentIdentifiedPrecursorIntensity"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["medianTicRtIqr"],"name":[25],"type":["dbl"],"align":["right"]},{"label":["medianTicOfRtRange"],"name":[26],"type":["dbl"],"align":["right"]},{"label":["mzAcquisitionRange.min"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["mzAcquisitionRange.max"],"name":[28],"type":["dbl"],"align":["right"]},{"label":["rtAcquisitionRange.min"],"name":[29],"type":["dbl"],"align":["right"]},{"label":["rtAcquisitionRange.max"],"name":[30],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityRange.min"],"name":[31],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityRange.max"],"name":[32],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q1"],"name":[33],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q2"],"name":[34],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q3"],"name":[35],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityMean"],"name":[36],"type":["dbl"],"align":["right"]},{"label":["precursorIntensitySd"],"name":[37],"type":["dbl"],"align":["right"]},{"label":["msSignal10xChange"],"name":[38],"type":["dbl"],"align":["right"]},{"label":["ratioCharge1over2"],"name":[39],"type":["dbl"],"align":["right"]},{"label":["ratioCharge3over2"],"name":[40],"type":["dbl"],"align":["right"]},{"label":["ratioCharge4over2"],"name":[41],"type":["dbl"],"align":["right"]},{"label":["meanCharge"],"name":[42],"type":["dbl"],"align":["right"]},{"label":["medianCharge"],"name":[43],"type":["dbl"],"align":["right"]}],"data":[{"1":"18.214","2":"0","3":"0.08098166","4":"0.08098166","5":"0.1495004","6":"1","7":"0.02893379","8":"0.08806413","9":"0.3102009","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"190","15":"0","16":"198.05","17":"5.10125","18":"18.42686","19":"655624525","20":"25612593","21":"389734297","22":"233673968","23":"5692460","24":"36467.88","25":"7496.006","26":"13858.935","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"100.6492356","32":"349282909","33":"1667.3911","34":"6746.195","35":"75003.90","36":"3450655.4","37":"26563228","38":"56","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.1_RPLC"},{"1":"18.214","2":"0","3":"0.14950038","4":"0.15202591","5":"0.1564181","6":"1","7":"0.03134951","8":"0.09031514","9":"0.3186011","10":"1","11":"-6.247847","12":"NaN","13":"NaN","14":"187","15":"0","16":"198.00","17":"5.14950","18":"18.06001","19":"1888825407","20":"25571443","21":"232310439","22":"1392480482","23":"238462898","24":"212376.68","25":"8418.277","26":"7641.905","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"100.8345421","32":"417130581","33":"918.6560","34":"6538.559","35":"106219.81","36":"10100670.6","37":"50739087","38":"50","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.2_RPLC"},{"1":"18.214","2":"0","3":"0.10645657","4":"0.14950038","5":"0.1564181","6":"1","7":"0.02860437","8":"0.09031514","9":"0.3186011","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"174","15":"0","16":"188.50","17":"5.24850","18":"16.38563","19":"1067244346","20":"13799650","21":"118319350","22":"863260007","23":"71865081","24":"100684.65","25":"10156.129","26":"14049.243","27":"60.1","28":"613.2","29":"0.986","30":"19.2","31":"203.5245443","32":"176368555","33":"1395.4894","34":"11330.716","35":"108566.82","36":"6133588.2","37":"27375719","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.3_RPLC"},{"1":"18.214","2":"0","3":"0.08098166","4":"0.14873174","5":"0.1564181","6":"1","7":"0.02860437","8":"0.08268365","9":"0.3069617","10":"1","11":"-5.213140","12":"NaN","13":"NaN","14":"185","15":"0","16":"197.00","17":"5.07000","18":"18.34320","19":"709060429","20":"58911195","21":"226139852","22":"315728188","23":"107539719","24":"65938.36","25":"12517.653","26":"24241.245","27":"60.1","28":"790.1","29":"0.986","30":"19.2","31":"200.2828695","32":"180664532","33":"1592.2577","34":"15757.838","35":"101354.21","36":"3832759.1","37":"17357083","38":"52","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.4_RPLC"},{"1":"18.214","2":"0","3":"0.10645657","4":"0.15202591","5":"0.3102009","6":"1","7":"0.03047107","8":"0.09031514","9":"0.3123971","10":"1","11":"-4.818931","12":"NaN","13":"NaN","14":"195","15":"0","16":"199.00","17":"5.11950","18":"18.94716","19":"1125965718","20":"34631015","21":"232611059","22":"652142304","23":"206571767","24":"226266.91","25":"6742.151","26":"6742.151","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"2.8536935","32":"186473049","33":"1281.4522","34":"7846.075","35":"80935.19","36":"5774183.2","37":"25295106","38":"61","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.5_RPLC"},{"1":"18.214","2":"0","3":"0.08098166","4":"0.08098166","5":"0.1520259","6":"1","7":"0.02860437","8":"0.08328758","9":"0.3069617","10":"1","11":"-7.875570","12":"NaN","13":"NaN","14":"188","15":"0","16":"197.55","17":"5.08025","18":"18.50303","19":"422868797","20":"12186285","21":"222228475","22":"154983024","23":"33301265","24":"16784.11","25":"44117.863","26":"33024.697","27":"60.1","28":"790.1","29":"0.986","30":"19.2","31":"100.0948557","32":"178131774","33":"2530.2667","34":"19178.494","35":"180318.36","36":"2249302.1","37":"13889398","38":"53","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.7_RPLC"},{"1":"18.214","2":"0","3":"0.03920061","4":"0.14950038","5":"0.3102009","6":"1","7":"0.02783573","8":"0.08048754","9":"0.3102009","10":"1","11":"-5.034679","12":"NaN","13":"NaN","14":"157","15":"0","16":"180.10","17":"5.14300","18":"15.36068","19":"69209907","20":"14116332","21":"9949038","22":"36036100","23":"9055468","24":"11684.77","25":"6760.000","26":"8176.000","27":"60.1","28":"664.1","29":"0.986","30":"19.2","31":"216.0000000","32":"9669053","33":"990.0000","34":"5746.000","35":"46548.00","36":"440827.4","37":"1401540","38":"43","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.8_RPLC"},{"1":"18.214","2":"0","3":"0.14873174","4":"0.17843417","5":"0.2387175","6":"1","7":"0.02893379","8":"0.08806413","9":"0.3102009","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"190","15":"0","16":"198.05","17":"5.10125","18":"18.42686","19":"393941299","20":"10471174","21":"73339944","22":"252118379","23":"57982267","24":"35693.70","25":"45252.398","26":"23497.922","27":"60.1","28":"790.1","29":"0.986","30":"19.2","31":"113.6090232","32":"100027496","33":"2314.9975","34":"17065.765","35":"103167.57","36":"2073375.3","37":"8851938","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.10_RPLC"},{"1":"18.214","2":"0","3":"0.08098166","4":"0.14950038","5":"0.1784342","6":"1","7":"0.02893379","8":"0.08696607","9":"0.3102009","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"190","15":"0","16":"198.05","17":"5.10125","18":"18.42686","19":"463158627","20":"19882709","21":"132373034","22":"214377252","23":"96500013","24":"35124.87","25":"9109.555","26":"16200.308","27":"60.1","28":"790.1","29":"0.986","30":"19.2","31":"111.5145088","32":"63873228","33":"2258.7437","34":"10061.898","35":"113136.70","36":"2437677.0","37":"8994466","38":"57","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.11_RPLC"},{"1":"18.214","2":"0","3":"0.09031514","4":"0.15202591","5":"0.3102009","6":"1","7":"0.03047107","8":"0.09031514","9":"0.3123971","10":"1","11":"-7.656151","12":"NaN","13":"NaN","14":"195","15":"0","16":"199.00","17":"5.11950","18":"18.94716","19":"1174772268","20":"34260238","21":"270065888","22":"709016110","23":"161424345","24":"264586.66","25":"4157.841","26":"5270.223","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"0.4717918","32":"193462858","33":"1291.5524","34":"5853.212","35":"84211.92","36":"6024473.2","37":"26285992","38":"56","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.12_RPLC"},{"1":"18.214","2":"0","3":"0.14950038","4":"0.15641814","5":"0.2387175","6":"1","7":"0.02893379","8":"0.08696607","9":"0.3069617","10":"1","11":"-7.459385","12":"NaN","13":"NaN","14":"190","15":"0","16":"198.05","17":"4.98925","18":"18.84051","19":"366273688","20":"14816072","21":"54751562","22":"233451189","23":"63215711","24":"26912.32","25":"38430.786","26":"25281.045","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"111.5123506","32":"77658882","33":"1503.4606","34":"12136.563","35":"125071.66","36":"1927756.3","37":"8078013","38":"54","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.14_RPLC"},{"1":"18.214","2":"0","3":"0.08054244","4":"0.14873174","5":"0.1564181","6":"1","7":"0.02893379","8":"0.08806413","9":"0.3115735","10":"1","11":"-7.525694","12":"NaN","13":"NaN","14":"195","15":"0","16":"199.00","17":"5.12150","18":"18.93976","19":"685406217","20":"85149119","21":"120417738","22":"453385872","23":"25746230","24":"157167.50","25":"12620.033","26":"21856.642","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"3.7525935","32":"85565295","33":"2255.9141","34":"15171.332","35":"177501.34","36":"3514903.7","37":"13108050","38":"61","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.15_RPLC"},{"1":"18.214","2":"0","3":"0.10645657","4":"0.15202591","5":"0.1564181","6":"1","7":"0.02860437","8":"0.08696607","9":"0.3115735","10":"1","11":"-6.580914","12":"NaN","13":"NaN","14":"183","15":"0","16":"197.00","17":"5.13850","18":"17.70945","19":"923692227","20":"13162898","21":"106462399","22":"734551101","23":"69513340","24":"74968.46","25":"8165.365","26":"9132.884","27":"60.1","28":"664.1","29":"0.986","30":"19.2","31":"203.2021021","32":"156599513","33":"1207.2282","34":"8174.571","35":"81675.65","36":"5047498.5","37":"22612632","38":"56","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.16_RPLC"},{"1":"18.214","2":"0","3":"0.08098166","4":"0.14950038","5":"0.3102009","6":"1","7":"0.02860437","8":"0.08328758","9":"0.3102009","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"192","15":"0","16":"198.05","17":"5.13075","18":"18.71071","19":"812167605","20":"71128784","21":"167377288","22":"470532428","23":"103061317","24":"79933.89","25":"39734.615","26":"50312.284","27":"60.1","28":"790.1","29":"0.986","30":"19.2","31":"100.2008199","32":"108004595","33":"2985.7138","34":"20803.475","35":"323337.46","36":"4230039.6","37":"15644017","38":"59","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.17_RPLC"},{"1":"18.214","2":"0","3":"0.10645657","4":"0.15202591","5":"0.3102009","6":"1","7":"0.02860437","8":"0.08696607","9":"0.2920830","10":"1","11":"-8.696981","12":"NaN","13":"NaN","14":"183","15":"0","16":"198.00","17":"4.73500","18":"19.21859","19":"420907302","20":"50534533","21":"45055320","22":"178649611","23":"146632732","24":"40120.41","25":"11322.484","26":"8129.251","27":"60.1","28":"664.1","29":"0.986","30":"19.2","31":"207.8654294","32":"82154906","33":"1786.1194","34":"11254.862","35":"106136.63","36":"2300039.9","37":"9400036","38":"49","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.18_RPLC"},{"1":"18.214","2":"0","3":"0.14950038","4":"0.15202591","5":"0.3102009","6":"1","7":"0.02893379","8":"0.09031514","9":"0.3123971","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"195","15":"0","16":"199.00","17":"5.14150","18":"18.86609","19":"1219274539","20":"31827447","21":"219412340","22":"787225580","23":"180801172","24":"232540.55","25":"4245.771","26":"4812.545","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"0.4225023","32":"212657623","33":"967.8666","34":"6292.957","35":"67140.32","36":"6252689.9","37":"28802825","38":"60","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.19_RPLC"},{"1":"18.214","2":"0","3":"0.06884814","4":"0.15202591","5":"0.3102009","6":"1","7":"0.02860437","8":"0.08696607","9":"0.3069617","10":"1","11":"-4.644165","12":"NaN","13":"NaN","14":"183","15":"0","16":"197.10","17":"4.93150","18":"18.45280","19":"378116398","20":"58009012","21":"50291892","22":"158344640","23":"111340634","24":"49532.61","25":"10973.490","26":"8586.033","27":"60.1","28":"664.1","29":"0.986","30":"19.2","31":"208.9190469","32":"66338080","33":"1802.5974","34":"9102.187","35":"122093.05","36":"2066209.8","37":"7965847","38":"48","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.20_RPLC"},{"1":"18.214","2":"0","3":"0.03683979","4":"0.06884814","5":"0.1495004","6":"1","7":"0.02860437","8":"0.08306797","9":"0.2920830","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"180","15":"0","16":"197.05","17":"4.86225","18":"18.50995","19":"367287695","20":"74686133","21":"163176742","22":"126815842","23":"2079079","24":"37989.54","25":"10409.833","26":"25253.263","27":"60.1","28":"664.1","29":"0.986","30":"19.2","31":"212.9997062","32":"59364514","33":"1241.9140","34":"9794.316","35":"182541.85","36":"2040487.2","37":"7577896","38":"52","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.21_RPLC"},{"1":"18.214","2":"0","3":"0.10645657","4":"0.14950038","5":"0.1564181","6":"1","7":"0.02893379","8":"0.09031514","9":"0.3186011","10":"1","11":"-6.220696","12":"NaN","13":"NaN","14":"177","15":"0","16":"191.00","17":"5.27600","18":"16.86884","19":"1084987782","20":"19690552","21":"177185730","22":"816287944","23":"71823289","24":"95662.89","25":"6577.034","26":"16562.811","27":"60.1","28":"613.2","29":"0.986","30":"19.2","31":"205.9864436","32":"176019325","33":"1260.4296","34":"10118.065","35":"105419.18","36":"6129874.5","37":"27034456","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.22_RPLC"},{"1":"18.214","2":"0","3":"0.14950038","4":"0.15202591","5":"0.1564181","6":"1","7":"0.02893379","8":"0.08696607","9":"0.3123971","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"184","15":"0","16":"196.55","17":"5.17025","18":"17.79411","19":"2004259781","20":"12182719","21":"219846849","22":"1532940458","23":"239288446","24":"128236.44","25":"10280.358","26":"8450.569","27":"60.1","28":"1377.6","29":"0.986","30":"19.2","31":"100.2840126","32":"418671634","33":"1308.6202","34":"8134.046","35":"140488.35","36":"10892716.2","37":"53948595","38":"48","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.23_RPLC"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre><code><span><span class="co">## [1] "metrics_hilic_msexp"</span></span></code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["chromatographyDuration"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.0%"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.25%"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.50%"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.75%"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["ticQuartersRtFraction.100%"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter1"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter2"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter3"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["rtOverMsQuarters.Quarter4"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q2/Q1"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q3/Q1"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["ticQuartileToQuartileLogRatio.Q4/Q1"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["numberSpectra"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["numberEmptyScans"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["medianPrecursorMz"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["rtIqr"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["rtIqrRate"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["areaUnderTic"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.25%"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.50%"],"name":[21],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.75%"],"name":[22],"type":["dbl"],"align":["right"]},{"label":["areaUnderTicRtQuantiles.100%"],"name":[23],"type":["dbl"],"align":["right"]},{"label":["extentIdentifiedPrecursorIntensity"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["medianTicRtIqr"],"name":[25],"type":["dbl"],"align":["right"]},{"label":["medianTicOfRtRange"],"name":[26],"type":["dbl"],"align":["right"]},{"label":["mzAcquisitionRange.min"],"name":[27],"type":["dbl"],"align":["right"]},{"label":["mzAcquisitionRange.max"],"name":[28],"type":["dbl"],"align":["right"]},{"label":["rtAcquisitionRange.min"],"name":[29],"type":["dbl"],"align":["right"]},{"label":["rtAcquisitionRange.max"],"name":[30],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityRange.min"],"name":[31],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityRange.max"],"name":[32],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q1"],"name":[33],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q2"],"name":[34],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityQuartiles.Q3"],"name":[35],"type":["dbl"],"align":["right"]},{"label":["precursorIntensityMean"],"name":[36],"type":["dbl"],"align":["right"]},{"label":["precursorIntensitySd"],"name":[37],"type":["dbl"],"align":["right"]},{"label":["msSignal10xChange"],"name":[38],"type":["dbl"],"align":["right"]},{"label":["ratioCharge1over2"],"name":[39],"type":["dbl"],"align":["right"]},{"label":["ratioCharge3over2"],"name":[40],"type":["dbl"],"align":["right"]},{"label":["ratioCharge4over2"],"name":[41],"type":["dbl"],"align":["right"]},{"label":["meanCharge"],"name":[42],"type":["dbl"],"align":["right"]},{"label":["medianCharge"],"name":[43],"type":["dbl"],"align":["right"]}],"data":[{"1":"16.0","2":"0","3":"0.34375000","4":"0.3437500","5":"0.5375000","6":"1","7":"0.1562500","8":"0.4750000","9":"0.6216875","10":"1","11":"-Inf","12":"NaN","13":"NaN","14":"165","15":"0","16":"179.10","17":"7.44700","18":"11.145428","19":"149945016","20":"927493.9","21":"100961764","22":"40996727","23":"5460065","24":"8713.9056","25":"2195.400","26":"2086.4171","27":"73","28":"784.1","29":"1.1","30":"17.1","31":"100.089504","32":"92021751","33":"300.50377","34":"2304.3834","35":"33382.253","36":"908757.67","37":"7729451","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.1_HILIC"},{"1":"14.9","2":"0","3":"0.00000000","4":"0.0000000","5":"0.3691275","6":"1","7":"0.1469799","8":"0.4428188","9":"0.6174497","10":"1","11":"-4.589484","12":"NaN","13":"NaN","14":"155","15":"0","16":"175.00","17":"6.90500","18":"11.151340","19":"123474393","20":"4465880.6","21":"28226161","22":"11799354","23":"1103962","24":"10755.1351","25":"2224.221","26":"1905.9756","27":"73","28":"784.1","29":"1.1","30":"16.0","31":"100.324179","32":"77879035","33":"331.89353","34":"1477.4019","35":"17697.319","36":"796608.99","37":"6406442","38":"38","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.2_HILIC"},{"1":"16.0","2":"0","3":"0.34375000","4":"0.4687500","5":"0.5312500","6":"1","7":"0.1475000","8":"0.4437500","9":"0.5903750","10":"1","11":"-9.657215","12":"NaN","13":"NaN","14":"157","15":"0","16":"174.10","17":"7.08600","18":"11.148744","19":"99234972","20":"659503.8","21":"34416012","22":"51698029","23":"12183170","24":"13710.3993","25":"4898.332","26":"4096.4119","27":"73","28":"662.1","29":"1.1","30":"17.1","31":"101.103201","32":"28026412","33":"298.60608","34":"1800.4271","35":"23690.197","36":"632069.88","37":"3265082","38":"38","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.3_HILIC"},{"1":"16.0","2":"0","3":"0.34375000","4":"0.4687500","5":"0.5312500","6":"1","7":"0.1562500","8":"0.4687500","9":"0.6187500","10":"1","11":"-Inf","12":"NaN","13":"NaN","14":"172","15":"0","16":"179.05","17":"7.41175","18":"11.873039","19":"143633524","20":"867006.7","21":"72103140","22":"51801242","23":"18126116","24":"19337.0677","25":"6423.865","26":"11595.1269","27":"73","28":"784.1","29":"1.1","30":"17.1","31":"105.754054","32":"45132582","33":"541.83574","34":"3876.1099","35":"53562.924","36":"835078.63","37":"4227102","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.4_HILIC"},{"1":"16.7","2":"0","3":"0.07784431","4":"0.2335329","5":"0.4910180","6":"1","7":"0.1377246","8":"0.4491018","9":"0.5956287","10":"1","11":"-6.565803","12":"NaN","13":"NaN","14":"192","15":"0","16":"181.05","17":"7.61525","18":"12.606283","19":"85968632","20":"40218490.4","21":"23447807","22":"5904716","23":"16271944","24":"124778.5950","25":"2019.404","26":"792.1483","27":"73","28":"808.1","29":"1.1","30":"17.8","31":"1.238693","32":"25393938","33":"59.17395","34":"1228.9237","35":"10633.082","36":"447753.29","37":"2316022","38":"58","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.5_HILIC"},{"1":"16.7","2":"0","3":"0.32934132","4":"0.3293413","5":"0.3293413","6":"1","7":"0.1497006","8":"0.4550898","9":"0.6008383","10":"1","11":"-6.388281","12":"NaN","13":"NaN","14":"173","15":"0","16":"184.00","17":"7.53400","18":"11.547651","19":"114112431","20":"2567409.3","21":"90088621","22":"15572740","23":"4899796","24":"7815.2363","25":"7310.729","26":"6141.5249","27":"73","28":"784.1","29":"1.1","30":"17.8","31":"101.938715","32":"76419185","33":"389.43830","34":"5265.1885","35":"39562.095","36":"659609.43","37":"5858571","38":"49","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.7_HILIC"},{"1":"16.0","2":"0","3":"0.34375000","4":"0.5312500","5":"0.7125000","6":"1","7":"0.1312500","8":"0.4703750","9":"0.6312500","10":"1","11":"-4.843906","12":"NaN","13":"NaN","14":"132","15":"0","16":"182.55","17":"8.02500","18":"8.348910","19":"3626196","20":"254109.0","21":"655525","22":"1440982","23":"1169178","24":"1788.6238","25":"1020.000","26":"775.5000","27":"89","28":"662.1","29":"1.1","30":"17.1","31":"54.000000","32":"1058811","33":"137.75000","34":"859.0000","35":"4303.250","36":"27471.18","37":"115061","38":"31","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.8_HILIC"},{"1":"16.0","2":"0","3":"0.34375000","4":"0.3437500","5":"0.3437500","6":"1","7":"0.1562500","8":"0.4687500","9":"0.6143750","10":"1","11":"-4.816047","12":"NaN","13":"NaN","14":"174","15":"0","16":"179.60","17":"7.32175","18":"11.882405","19":"207066273","20":"12353340.6","21":"146931805","22":"21497066","23":"6167214","24":"22996.4252","25":"2328.448","26":"1571.0124","27":"73","28":"784.1","29":"1.1","30":"17.1","31":"100.188629","32":"128106182","33":"370.57743","34":"1678.0164","35":"19839.197","36":"1190036.05","37":"9898275","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.10_HILIC"},{"1":"16.7","2":"0","3":"0.32934132","4":"0.4416168","5":"0.5209581","6":"1","7":"0.1497006","8":"0.4550898","9":"0.5988024","10":"1","11":"-Inf","12":"NaN","13":"NaN","14":"176","15":"0","16":"183.55","17":"7.50000","18":"12.000000","19":"90010103","20":"429874.4","21":"46882993","22":"29142937","23":"12534957","24":"11350.2701","25":"17582.482","26":"25942.8704","27":"73","28":"808.1","29":"1.1","30":"17.8","31":"106.097318","32":"34014953","33":"464.99864","34":"6462.0250","35":"58110.025","36":"511421.04","37":"2791495","38":"47","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.11_HILIC"},{"1":"16.7","2":"0","3":"0.11976048","4":"0.2395210","5":"0.5089820","6":"1","7":"0.1413174","8":"0.4506587","9":"0.5988024","10":"1","11":"NaN","12":"NaN","13":"NaN","14":"191","15":"0","16":"183.10","17":"7.57000","18":"12.681638","19":"102477573","20":"30166449.2","21":"41097905","22":"15329409","23":"15666633","24":"122437.3333","25":"1501.044","26":"555.1991","27":"73","28":"808.1","29":"1.1","30":"17.8","31":"1.433023","32":"22618425","33":"78.80922","34":"983.1723","35":"8181.036","36":"536531.79","37":"2617586","38":"51","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.12_HILIC"},{"1":"16.7","2":"0","3":"0.32934132","4":"0.3293413","5":"0.3293413","6":"1","7":"0.1331737","8":"0.4131737","9":"0.5664671","10":"1","11":"-4.047178","12":"NaN","13":"NaN","14":"160","15":"0","16":"176.05","17":"7.18900","18":"11.128112","19":"197740157","20":"12009691.6","21":"132437662","22":"30998112","23":"1463422","24":"20138.7341","25":"1246.187","26":"1103.2935","27":"73","28":"784.1","29":"1.1","30":"17.8","31":"100.659249","32":"114631019","33":"310.26870","34":"1259.1343","35":"15722.035","36":"1235875.98","37":"9311172","38":"37","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.14_HILIC"},{"1":"16.0","2":"0","3":"0.08125000","4":"0.3687500","5":"0.6216875","6":"1","7":"0.1475000","8":"0.4687500","9":"0.6187500","10":"1","11":"-Inf","12":"NaN","13":"NaN","14":"179","15":"0","16":"180.10","17":"7.43500","18":"11.970410","19":"102361753","20":"4139131.0","21":"28600788","22":"20071898","23":"26364040","24":"20161.5384","25":"6657.592","26":"10877.6175","27":"73","28":"808.1","29":"1.1","30":"17.1","31":"100.227835","32":"23185896","33":"559.37276","34":"5958.4497","35":"79191.771","36":"571853.37","37":"2363773","38":"51","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.15_HILIC"},{"1":"16.0","2":"0","3":"0.24375000","4":"0.4687500","5":"0.5312500","6":"1","7":"0.1437500","8":"0.4437500","9":"0.5912500","10":"1","11":"-5.873030","12":"NaN","13":"NaN","14":"153","15":"0","16":"173.10","17":"7.16000","18":"10.754190","19":"91545940","20":"772362.2","21":"27315566","22":"52075604","23":"11038000","24":"7247.6642","25":"6517.689","26":"6222.8088","27":"73","28":"662.1","29":"1.1","30":"17.1","31":"100.052023","32":"22774034","33":"421.51655","34":"2228.2204","35":"23912.830","36":"598339.48","37":"3056180","38":"37","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.16_HILIC"},{"1":"16.7","2":"0","3":"0.32934132","4":"0.4970060","5":"0.5808383","6":"1","7":"0.1497006","8":"0.4491018","9":"0.5956287","10":"1","11":"-8.794215","12":"NaN","13":"NaN","14":"180","15":"0","16":"179.10","17":"7.46025","18":"12.197983","19":"462124450","20":"33772023.7","21":"188895639","22":"156828313","23":"76947218","24":"48192.2050","25":"17778.645","26":"40107.3763","27":"73","28":"784.1","29":"1.1","30":"17.8","31":"112.047959","32":"116592442","33":"1623.13611","34":"15994.4296","35":"573235.872","36":"2567358.06","37":"10990394","38":"53","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.17_HILIC"},{"1":"14.6","2":"0","3":"0.37671233","4":"0.3767123","5":"0.5684932","6":"1","7":"0.2739726","8":"0.5547945","9":"0.6986301","10":"1","11":"-4.990415","12":"NaN","13":"NaN","14":"99","15":"0","16":"182.00","17":"6.20000","18":"8.225806","19":"49453064","20":"8302314.0","21":"24977590","22":"12861026","23":"3109587","24":"1141.4487","25":"18706.986","26":"11766.0195","27":"88","28":"784.1","29":"1.1","30":"15.7","31":"638.351172","32":"19551920","33":"3856.80632","34":"12591.5782","35":"114968.303","36":"499525.90","37":"2159278","38":"21","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.18_HILIC"},{"1":"16.7","2":"0","3":"0.07784431","4":"0.2395210","5":"0.5089820","6":"1","7":"0.1497006","8":"0.4550898","9":"0.5988024","10":"1","11":"-7.401569","12":"NaN","13":"NaN","14":"189","15":"0","16":"182.00","17":"7.50000","18":"12.800000","19":"98117684","20":"30246056.5","21":"33400459","22":"15286939","23":"19074306","24":"139782.5632","25":"1550.186","26":"728.3760","27":"73","28":"808.1","29":"1.1","30":"17.8","31":"1.524586","32":"25312088","33":"92.13823","34":"1029.8871","35":"10296.071","36":"519141.19","37":"2654500","38":"57","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.19_HILIC"},{"1":"14.5","2":"0","3":"0.37931034","4":"0.3793103","5":"0.5724138","6":"1","7":"0.2758621","8":"0.5586207","9":"0.7103448","10":"1","11":"-3.881831","12":"NaN","13":"NaN","14":"96","15":"0","16":"174.05","17":"6.30000","18":"7.936508","19":"47908600","20":"6969702.6","21":"27633400","22":"11093074","23":"2125419","24":"820.0707","25":"21810.222","26":"12476.0812","27":"88","28":"784.1","29":"1.1","30":"15.6","31":"578.762163","32":"22314168","33":"4366.68909","34":"12087.4803","35":"109054.742","36":"499047.92","37":"2403802","38":"22","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.20_HILIC"},{"1":"14.6","2":"0","3":"0.37671233","4":"0.3767123","5":"0.3767123","6":"1","7":"0.2260274","8":"0.5547945","9":"0.6986301","10":"1","11":"-7.290321","12":"NaN","13":"NaN","14":"117","15":"0","16":"188.00","17":"6.90000","18":"8.695652","19":"110479638","20":"20975351.2","21":"73424523","22":"12037342","23":"3754587","24":"1341.2704","25":"41146.272","26":"47912.7459","27":"88","28":"784.1","29":"1.1","30":"15.7","31":"1192.667987","32":"59944875","33":"3964.68826","34":"23221.5070","35":"287834.303","36":"944270.41","37":"5640411","38":"26","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.21_HILIC"},{"1":"16.0","2":"0","3":"0.24375000","4":"0.4687500","5":"0.5312500","6":"1","7":"0.1437500","8":"0.4437500","9":"0.5937500","10":"1","11":"-Inf","12":"NaN","13":"NaN","14":"162","15":"0","16":"174.05","17":"7.17500","18":"11.149826","19":"127306936","20":"14108553.0","21":"30188657","22":"68940112","23":"13835313","24":"16964.7330","25":"5054.809","26":"1998.6781","27":"73","28":"662.1","29":"1.1","30":"17.1","31":"101.247677","32":"30271949","33":"316.68560","34":"1769.8449","35":"23466.103","36":"785845.29","37":"3585511","38":"46","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.22_HILIC"},{"1":"14.9","2":"0","3":"0.13308725","4":"0.2617450","5":"0.3691275","6":"1","7":"0.1469799","8":"0.4765101","9":"0.6174497","10":"1","11":"-4.847856","12":"NaN","13":"NaN","14":"150","15":"0","16":"174.05","17":"7.00750","18":"10.702818","19":"98409667","20":"16122600.7","21":"51697900","22":"11930335","23":"1055401","24":"18937.7002","25":"3711.517","26":"2344.9065","27":"73","28":"611.1","29":"1.1","30":"16.0","31":"100.805934","32":"23916974","33":"440.73392","34":"1785.2185","35":"22085.106","36":"656064.44","37":"2813919","38":"34","39":"NaN","40":"NaN","41":"NaN","42":"NaN","43":"NA","_rn_":"Sample.23_HILIC"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-the-results">Visualizing the results<a class="anchor" aria-label="anchor" href="#visualizing-the-results"></a>
</h2>
<p>The quality metrics can be most easily compared when graphically
visualized.</p>
<p>The <code>MsQuality</code> package offers the possibility to
graphically display the metrics using the <code>plotMetric</code> and
<code>shinyMsQuality</code> functions. The <code>plotMetric</code>
function will create one plot based on a single metric.
<code>shinyMsQuality</code>, on the other hand, opens a shiny
application that allows to browse through all the metrics stored in the
object.</p>
<p>As a way of example, we will plot here the number of features. A high
number of missing features might indicate low data quality, however,
also different sample types might exhibit contrasting number of detected
features. As a general rule, only samples of the same type should be
compared to adjust for sample type-specific effects.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metrics_msexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">metrics_rplc_msexp</span>, <span class="va">metrics_hilic_msexp</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotMetric.html">plotMetric</a></span><span class="op">(</span>qc <span class="op">=</span> <span class="va">metrics_msexp</span>, metric <span class="op">=</span> <span class="st">"numberSpectra"</span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4ffd6be222d965d70fe0" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-4ffd6be222d965d70fe0">{"x":{"data":[{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[190,187,174,185,195,188,157,190,190,195,190,195,183,192,183,195,183,180,177,184,165,155,157,172,192,173,132,174,176,191,160,179,153,180,99,189,96,117,162,150],"text":["get(\"rowname\"): Sample.1_RPLC<br />get(\"value\"): 190","get(\"rowname\"): Sample.2_RPLC<br />get(\"value\"): 187","get(\"rowname\"): Sample.3_RPLC<br />get(\"value\"): 174","get(\"rowname\"): Sample.4_RPLC<br />get(\"value\"): 185","get(\"rowname\"): Sample.5_RPLC<br />get(\"value\"): 195","get(\"rowname\"): Sample.7_RPLC<br />get(\"value\"): 188","get(\"rowname\"): Sample.8_RPLC<br />get(\"value\"): 157","get(\"rowname\"): Sample.10_RPLC<br />get(\"value\"): 190","get(\"rowname\"): Sample.11_RPLC<br />get(\"value\"): 190","get(\"rowname\"): Sample.12_RPLC<br />get(\"value\"): 195","get(\"rowname\"): Sample.14_RPLC<br />get(\"value\"): 190","get(\"rowname\"): Sample.15_RPLC<br />get(\"value\"): 195","get(\"rowname\"): Sample.16_RPLC<br />get(\"value\"): 183","get(\"rowname\"): Sample.17_RPLC<br />get(\"value\"): 192","get(\"rowname\"): Sample.18_RPLC<br />get(\"value\"): 183","get(\"rowname\"): Sample.19_RPLC<br />get(\"value\"): 195","get(\"rowname\"): Sample.20_RPLC<br />get(\"value\"): 183","get(\"rowname\"): Sample.21_RPLC<br />get(\"value\"): 180","get(\"rowname\"): Sample.22_RPLC<br />get(\"value\"): 177","get(\"rowname\"): Sample.23_RPLC<br />get(\"value\"): 184","get(\"rowname\"): Sample.1_HILIC<br />get(\"value\"): 165","get(\"rowname\"): Sample.2_HILIC<br />get(\"value\"): 155","get(\"rowname\"): Sample.3_HILIC<br />get(\"value\"): 157","get(\"rowname\"): Sample.4_HILIC<br />get(\"value\"): 172","get(\"rowname\"): Sample.5_HILIC<br />get(\"value\"): 192","get(\"rowname\"): Sample.7_HILIC<br />get(\"value\"): 173","get(\"rowname\"): Sample.8_HILIC<br />get(\"value\"): 132","get(\"rowname\"): Sample.10_HILIC<br />get(\"value\"): 174","get(\"rowname\"): Sample.11_HILIC<br />get(\"value\"): 176","get(\"rowname\"): Sample.12_HILIC<br />get(\"value\"): 191","get(\"rowname\"): Sample.14_HILIC<br />get(\"value\"): 160","get(\"rowname\"): Sample.15_HILIC<br />get(\"value\"): 179","get(\"rowname\"): Sample.16_HILIC<br />get(\"value\"): 153","get(\"rowname\"): Sample.17_HILIC<br />get(\"value\"): 180","get(\"rowname\"): Sample.18_HILIC<br />get(\"value\"):  99","get(\"rowname\"): Sample.19_HILIC<br />get(\"value\"): 189","get(\"rowname\"): Sample.20_HILIC<br />get(\"value\"):  96","get(\"rowname\"): Sample.21_HILIC<br />get(\"value\"): 117","get(\"rowname\"): Sample.22_HILIC<br />get(\"value\"): 162","get(\"rowname\"): Sample.23_HILIC<br />get(\"value\"): 150"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(228,26,28,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(228,26,28,1)"}},"hoveron":"points","name":"numberSpectra","legendgroup":"numberSpectra","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":45.358882994166862,"r":7.3059360730593621,"b":129.71587759773678,"l":43.105022831050235},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"title":{"text":"numberSpectra","font":{"color":"rgba(0,0,0,1)","family":"","size":17.534246575342465},"x":0,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,40.600000000000001],"tickmode":"array","ticktext":["Sample.1_RPLC","Sample.2_RPLC","Sample.3_RPLC","Sample.4_RPLC","Sample.5_RPLC","Sample.7_RPLC","Sample.8_RPLC","Sample.10_RPLC","Sample.11_RPLC","Sample.12_RPLC","Sample.14_RPLC","Sample.15_RPLC","Sample.16_RPLC","Sample.17_RPLC","Sample.18_RPLC","Sample.19_RPLC","Sample.20_RPLC","Sample.21_RPLC","Sample.22_RPLC","Sample.23_RPLC","Sample.1_HILIC","Sample.2_HILIC","Sample.3_HILIC","Sample.4_HILIC","Sample.5_HILIC","Sample.7_HILIC","Sample.8_HILIC","Sample.10_HILIC","Sample.11_HILIC","Sample.12_HILIC","Sample.14_HILIC","Sample.15_HILIC","Sample.16_HILIC","Sample.17_HILIC","Sample.18_HILIC","Sample.19_HILIC","Sample.20_HILIC","Sample.21_HILIC","Sample.22_HILIC","Sample.23_HILIC"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"categoryorder":"array","categoryarray":["Sample.1_RPLC","Sample.2_RPLC","Sample.3_RPLC","Sample.4_RPLC","Sample.5_RPLC","Sample.7_RPLC","Sample.8_RPLC","Sample.10_RPLC","Sample.11_RPLC","Sample.12_RPLC","Sample.14_RPLC","Sample.15_RPLC","Sample.16_RPLC","Sample.17_RPLC","Sample.18_RPLC","Sample.19_RPLC","Sample.20_RPLC","Sample.21_RPLC","Sample.22_RPLC","Sample.23_RPLC","Sample.1_HILIC","Sample.2_HILIC","Sample.3_HILIC","Sample.4_HILIC","Sample.5_HILIC","Sample.7_HILIC","Sample.8_HILIC","Sample.10_HILIC","Sample.11_HILIC","Sample.12_HILIC","Sample.14_HILIC","Sample.15_HILIC","Sample.16_HILIC","Sample.17_HILIC","Sample.18_HILIC","Sample.19_HILIC","Sample.20_HILIC","Sample.21_HILIC","Sample.22_HILIC","Sample.23_HILIC"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.283520132835198},"tickangle":-90,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"sample","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[91.049999999999997,199.94999999999999],"tickmode":"array","ticktext":["100","125","150","175"],"tickvals":[100,125,150,175],"categoryorder":"array","categoryarray":["100","125","150","175"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"get(\"value\")","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"get(\"name\")","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"1644328cc545":{"x":{},"y":{},"colour":{},"type":"scatter"}},"cur_data":"1644328cc545","visdat":{"1644328cc545":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>Similarly, we are able to display the area under the TIC for the
retention time quantiles. This plot gives information on the perceived
signal (TIC) for the differnt retention time quantiles and could
indicate drifts or interruptions of sensitivity during the run.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotMetric.html">plotMetric</a></span><span class="op">(</span>qc <span class="op">=</span> <span class="va">metrics_msexp</span>, metric <span class="op">=</span> <span class="st">"ticQuartileToQuartileLogRatio"</span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-cda8c601604ed98db196" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-cda8c601604ed98db196">{"x":{"data":[{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[null,-6.247847415939181,null,-5.2131404597468967,-4.8189305255858423,-7.8755700833394746,-5.0346787478350885,null,null,-7.6561514260073897,-7.4593845701252368,-7.5256935216518626,-6.580914449316742,null,-8.6969814566978805,null,-4.6441652702008565,null,-6.2206957660759947,null,null,-4.5894835136640397,-9.6572149221986443,null,-6.5658029807260272,-6.3882805958497846,-4.8439058320621369,-4.816047130432378,null,null,-4.0471782006449688,null,-5.8730296615302056,-8.7942147376256123,-4.9904150526104418,-7.4015692099381489,-3.8818310959632596,-7.2903214507605476,null,-4.847856197124675],"text":["get(\"rowname\"): Sample.1_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.2_RPLC<br />get(\"value\"): -6.247847","get(\"rowname\"): Sample.3_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.4_RPLC<br />get(\"value\"): -5.213140","get(\"rowname\"): Sample.5_RPLC<br />get(\"value\"): -4.818931","get(\"rowname\"): Sample.7_RPLC<br />get(\"value\"): -7.875570","get(\"rowname\"): Sample.8_RPLC<br />get(\"value\"): -5.034679","get(\"rowname\"): Sample.10_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.11_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.12_RPLC<br />get(\"value\"): -7.656151","get(\"rowname\"): Sample.14_RPLC<br />get(\"value\"): -7.459385","get(\"rowname\"): Sample.15_RPLC<br />get(\"value\"): -7.525694","get(\"rowname\"): Sample.16_RPLC<br />get(\"value\"): -6.580914","get(\"rowname\"): Sample.17_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.18_RPLC<br />get(\"value\"): -8.696981","get(\"rowname\"): Sample.19_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.20_RPLC<br />get(\"value\"): -4.644165","get(\"rowname\"): Sample.21_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.22_RPLC<br />get(\"value\"): -6.220696","get(\"rowname\"): Sample.23_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.1_HILIC<br />get(\"value\"):      -Inf","get(\"rowname\"): Sample.2_HILIC<br />get(\"value\"): -4.589484","get(\"rowname\"): Sample.3_HILIC<br />get(\"value\"): -9.657215","get(\"rowname\"): Sample.4_HILIC<br />get(\"value\"):      -Inf","get(\"rowname\"): Sample.5_HILIC<br />get(\"value\"): -6.565803","get(\"rowname\"): Sample.7_HILIC<br />get(\"value\"): -6.388281","get(\"rowname\"): Sample.8_HILIC<br />get(\"value\"): -4.843906","get(\"rowname\"): Sample.10_HILIC<br />get(\"value\"): -4.816047","get(\"rowname\"): Sample.11_HILIC<br />get(\"value\"):      -Inf","get(\"rowname\"): Sample.12_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.14_HILIC<br />get(\"value\"): -4.047178","get(\"rowname\"): Sample.15_HILIC<br />get(\"value\"):      -Inf","get(\"rowname\"): Sample.16_HILIC<br />get(\"value\"): -5.873030","get(\"rowname\"): Sample.17_HILIC<br />get(\"value\"): -8.794215","get(\"rowname\"): Sample.18_HILIC<br />get(\"value\"): -4.990415","get(\"rowname\"): Sample.19_HILIC<br />get(\"value\"): -7.401569","get(\"rowname\"): Sample.20_HILIC<br />get(\"value\"): -3.881831","get(\"rowname\"): Sample.21_HILIC<br />get(\"value\"): -7.290321","get(\"rowname\"): Sample.22_HILIC<br />get(\"value\"):      -Inf","get(\"rowname\"): Sample.23_HILIC<br />get(\"value\"): -4.847856"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(228,26,28,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(228,26,28,1)"}},"hoveron":"points","name":".Q2/Q1","legendgroup":".Q2/Q1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"text":["get(\"rowname\"): Sample.1_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.2_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.3_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.4_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.5_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.7_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.8_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.10_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.11_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.12_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.14_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.15_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.16_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.17_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.18_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.19_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.20_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.21_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.22_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.23_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.1_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.2_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.3_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.4_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.5_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.7_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.8_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.10_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.11_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.12_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.14_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.15_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.16_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.17_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.18_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.19_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.20_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.21_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.22_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.23_HILIC<br />get(\"value\"):       NaN"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(55,126,184,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(55,126,184,1)"}},"hoveron":"points","name":".Q3/Q1","legendgroup":".Q3/Q1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"y":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"text":["get(\"rowname\"): Sample.1_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.2_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.3_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.4_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.5_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.7_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.8_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.10_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.11_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.12_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.14_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.15_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.16_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.17_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.18_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.19_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.20_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.21_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.22_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.23_RPLC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.1_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.2_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.3_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.4_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.5_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.7_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.8_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.10_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.11_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.12_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.14_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.15_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.16_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.17_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.18_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.19_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.20_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.21_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.22_HILIC<br />get(\"value\"):       NaN","get(\"rowname\"): Sample.23_HILIC<br />get(\"value\"):       NaN"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(77,175,74,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(77,175,74,1)"}},"hoveron":"points","name":".Q4/Q1","legendgroup":".Q4/Q1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":45.358882994166862,"r":7.3059360730593621,"b":129.71587759773678,"l":37.260273972602747},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"title":{"text":"ticQuartileToQuartileLogRatio","font":{"color":"rgba(0,0,0,1)","family":"","size":17.534246575342465},"x":0,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.40000000000000002,40.600000000000001],"tickmode":"array","ticktext":["Sample.1_RPLC","Sample.2_RPLC","Sample.3_RPLC","Sample.4_RPLC","Sample.5_RPLC","Sample.7_RPLC","Sample.8_RPLC","Sample.10_RPLC","Sample.11_RPLC","Sample.12_RPLC","Sample.14_RPLC","Sample.15_RPLC","Sample.16_RPLC","Sample.17_RPLC","Sample.18_RPLC","Sample.19_RPLC","Sample.20_RPLC","Sample.21_RPLC","Sample.22_RPLC","Sample.23_RPLC","Sample.1_HILIC","Sample.2_HILIC","Sample.3_HILIC","Sample.4_HILIC","Sample.5_HILIC","Sample.7_HILIC","Sample.8_HILIC","Sample.10_HILIC","Sample.11_HILIC","Sample.12_HILIC","Sample.14_HILIC","Sample.15_HILIC","Sample.16_HILIC","Sample.17_HILIC","Sample.18_HILIC","Sample.19_HILIC","Sample.20_HILIC","Sample.21_HILIC","Sample.22_HILIC","Sample.23_HILIC"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"categoryorder":"array","categoryarray":["Sample.1_RPLC","Sample.2_RPLC","Sample.3_RPLC","Sample.4_RPLC","Sample.5_RPLC","Sample.7_RPLC","Sample.8_RPLC","Sample.10_RPLC","Sample.11_RPLC","Sample.12_RPLC","Sample.14_RPLC","Sample.15_RPLC","Sample.16_RPLC","Sample.17_RPLC","Sample.18_RPLC","Sample.19_RPLC","Sample.20_RPLC","Sample.21_RPLC","Sample.22_RPLC","Sample.23_RPLC","Sample.1_HILIC","Sample.2_HILIC","Sample.3_HILIC","Sample.4_HILIC","Sample.5_HILIC","Sample.7_HILIC","Sample.8_HILIC","Sample.10_HILIC","Sample.11_HILIC","Sample.12_HILIC","Sample.14_HILIC","Sample.15_HILIC","Sample.16_HILIC","Sample.17_HILIC","Sample.18_HILIC","Sample.19_HILIC","Sample.20_HILIC","Sample.21_HILIC","Sample.22_HILIC","Sample.23_HILIC"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.283520132835198},"tickangle":-90,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"sample","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-9.9459841135104128,-3.5930619046514902],"tickmode":"array","ticktext":["-8","-6","-4"],"tickvals":[-8,-6,-4],"categoryorder":"array","categoryarray":["-8","-6","-4"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"get(\"value\")","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"get(\"name\")","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"16442c7c946f":{"x":{},"y":{},"colour":{},"type":"scatter"}},"cur_data":"16442c7c946f","visdat":{"16442c7c946f":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>Alternatively, to browse through all metrics that were calculated in
an interactive way, we can use the <code>shinyMsQuality</code>
function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/shinyMsQuality.html">shinyMsQuality</a></span><span class="op">(</span>qc <span class="op">=</span> <span class="va">metrics_msexp</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="appendix">Appendix<a class="anchor" aria-label="anchor" href="#appendix"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h3>
<p>All software and respective versions to build this vignette are
listed here:</p>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] MsExperiment_1.2.0  Spectra_1.10.2      ProtGenerics_1.32.0</span></span>
<span><span class="co">## [4] BiocParallel_1.34.2 S4Vectors_0.38.1    BiocGenerics_0.46.0</span></span>
<span><span class="co">## [7] MsQuality_1.1.2     knitr_1.43          BiocStyle_2.28.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] bitops_1.0-7                testthat_3.1.10            </span></span>
<span><span class="co">##   [3] rlang_1.1.1                 magrittr_2.0.3             </span></span>
<span><span class="co">##   [5] shinydashboard_0.7.2        clue_0.3-64                </span></span>
<span><span class="co">##   [7] matrixStats_1.0.0           compiler_4.3.1             </span></span>
<span><span class="co">##   [9] systemfonts_1.0.4           vctrs_0.6.3                </span></span>
<span><span class="co">##  [11] stringr_1.5.0               pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [13] crayon_1.5.2                fastmap_1.1.1              </span></span>
<span><span class="co">##  [15] ellipsis_0.3.2              XVector_0.40.0             </span></span>
<span><span class="co">##  [17] labeling_0.4.2              utf8_1.2.3                 </span></span>
<span><span class="co">##  [19] promises_1.2.1              rmarkdown_2.23             </span></span>
<span><span class="co">##  [21] ragg_1.2.5                  purrr_1.0.2                </span></span>
<span><span class="co">##  [23] xfun_0.40                   MultiAssayExperiment_1.26.0</span></span>
<span><span class="co">##  [25] zlibbioc_1.46.0             cachem_1.0.8               </span></span>
<span><span class="co">##  [27] GenomeInfoDb_1.36.1         jsonlite_1.8.7             </span></span>
<span><span class="co">##  [29] later_1.3.1                 DelayedArray_0.26.7        </span></span>
<span><span class="co">##  [31] parallel_4.3.1              cluster_2.1.4              </span></span>
<span><span class="co">##  [33] R6_2.5.1                    RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [35] bslib_0.5.0                 stringi_1.7.12             </span></span>
<span><span class="co">##  [37] brio_1.1.3                  GenomicRanges_1.52.0       </span></span>
<span><span class="co">##  [39] jquerylib_0.1.4             Rcpp_1.0.11                </span></span>
<span><span class="co">##  [41] bookdown_0.35               SummarizedExperiment_1.30.2</span></span>
<span><span class="co">##  [43] IRanges_2.34.1              httpuv_1.6.11              </span></span>
<span><span class="co">##  [45] Matrix_1.6-0                igraph_1.5.1               </span></span>
<span><span class="co">##  [47] tidyselect_1.2.0            abind_1.4-5                </span></span>
<span><span class="co">##  [49] yaml_2.3.7                  codetools_0.2-19           </span></span>
<span><span class="co">##  [51] curl_5.0.1                  lattice_0.21-8             </span></span>
<span><span class="co">##  [53] tibble_3.2.1                withr_2.5.0                </span></span>
<span><span class="co">##  [55] Biobase_2.60.0              shiny_1.7.4.1              </span></span>
<span><span class="co">##  [57] evaluate_0.21               ontologyIndex_2.11         </span></span>
<span><span class="co">##  [59] desc_1.4.2                  pillar_1.9.0               </span></span>
<span><span class="co">##  [61] BiocManager_1.30.22         MatrixGenerics_1.12.3      </span></span>
<span><span class="co">##  [63] plotly_4.10.2               generics_0.1.3             </span></span>
<span><span class="co">##  [65] rprojroot_2.0.3             RCurl_1.98-1.12            </span></span>
<span><span class="co">##  [67] ggplot2_3.4.2               munsell_0.5.0              </span></span>
<span><span class="co">##  [69] scales_1.2.1                R6P_0.3.0                  </span></span>
<span><span class="co">##  [71] xtable_1.8-4                glue_1.6.2                 </span></span>
<span><span class="co">##  [73] lazyeval_0.2.2              tools_4.3.1                </span></span>
<span><span class="co">##  [75] data.table_1.14.8           QFeatures_1.10.0           </span></span>
<span><span class="co">##  [77] fs_1.6.3                    grid_4.3.1                 </span></span>
<span><span class="co">##  [79] jsonvalidate_1.3.2          tidyr_1.3.0                </span></span>
<span><span class="co">##  [81] crosstalk_1.2.0             MsCoreUtils_1.12.0         </span></span>
<span><span class="co">##  [83] msdata_0.40.0               colorspace_2.1-0           </span></span>
<span><span class="co">##  [85] GenomeInfoDbData_1.2.10     cli_3.6.1                  </span></span>
<span><span class="co">##  [87] textshaping_0.3.6           fansi_1.0.4                </span></span>
<span><span class="co">##  [89] S4Arrays_1.0.5              viridisLite_0.4.2          </span></span>
<span><span class="co">##  [91] dplyr_1.1.2                 AnnotationFilter_1.24.0    </span></span>
<span><span class="co">##  [93] gtable_0.3.3                sass_0.4.7                 </span></span>
<span><span class="co">##  [95] digest_0.6.33               htmlwidgets_1.6.2          </span></span>
<span><span class="co">##  [97] memoise_2.0.1               htmltools_0.5.6            </span></span>
<span><span class="co">##  [99] pkgdown_2.0.7               lifecycle_1.0.3            </span></span>
<span><span class="co">## [101] httr_1.4.6                  mime_0.12                  </span></span>
<span><span class="co">## [103] rmzqc_0.5.2                 MASS_7.3-60</span></span></code></pre>
</div>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Lee2019" class="csl-entry">
Lee, H.-J., D. M. Kremer, P. Sajjakulnukit, L. Zhang, and C. A.
Lyssiotis. 2019. <span>“A Large-Scale Analysis of Targeted Metabolomics
Data from Heterogeneous Biological Samples Provides Insights into
Metabolite Dynamics.”</span> <em>Metabolomics</em>, 103. <a href="https://doi.org/10.1007/s11306-019-1564-8" class="external-link">https://doi.org/10.1007/s11306-019-1564-8</a>.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Thomas Naake, Johannes Rainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
